// @whiskeysockets/baileys/lib/WAM/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FLAG_EXTENDED = exports.FLAG_FIELD = exports.FLAG_EVENT = exports.FLAG_GLOBAL = exports.FLAG_BYTE = exports.WEB_GLOBALS = exports.WEB_EVENTS = void 0;
exports.WEB_EVENTS = [
    {
        name: 'WebcFingerprint',
        id: 1704,
        props: {
            webcCanvasFingerprint: [25, 'string'],
            webcDocumentDriverEvaluate: [9, 'boolean'],
            webcDocumentDriverUnwrapped: [7, 'boolean'],
            webcDocumentElementAttrDriver: [14, 'boolean'],
            webcDocumentElementAttrSelenium: [12, 'boolean'],
            webcDocumentElementAttrWebdriver: [13, 'boolean'],
            webcDocumentFxdriverEvaluate: [6, 'boolean'],
            webcDocumentFxdriverUnwrapped: [11, 'boolean'],
            webcDocumentSeleniumEvaluate: [2, 'boolean'],
            webcDocumentSeleniumUnwrapped: [10, 'boolean'],
            webcDocumentWebdriverEvaluate: [1, 'boolean'],
            webcDocumentWebdriverScriptFn: [5, 'boolean'],
            webcDocumentWebdriverScriptFunc: [4, 'boolean'],
            webcDocumentWebdriverScriptFunction: [3, 'boolean'],
            webcDocumentWebdriverUnwrapped: [8, 'boolean'],
            webcWebglFingerprint: [24, 'string'],
            webcWebglRenderer: [23, 'string'],
            webcWebglVendor: [22, 'string'],
            webcWindowSeleniumIdeRecorder: [20, 'boolean'],
            webcWindowCallphantom: [18, 'boolean'],
            webcWindowCallselenium: [19, 'boolean'],
            webcWindowNavigatorWebdriver: [
                21,
                {
                    FALSE: 0,
                    TRUE: 1,
                    UNDEFINED: 2,
                },
            ],
            webcWindowNightmare: [16, 'boolean'],
            webcWindowPhantom: [15, 'boolean'],
            webcWindowSelenium: [17, 'boolean'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'Login',
        id: 460,
        props: {
            androidKeystoreState: [
                10,
                {
                    NOT_AVAILABLE: 1,
                    SELF_TEST_FAILURE: 2,
                    ENC_KEY_READ_FAILURE: 3,
                    ENC_KEY_STORED_USED: 4,
                    ENC_KEY_PLAIN_DELETED: 5,
                    ENC_KEY_PLAIN_RECOVERED: 6,
                },
            ],
            connectionOrigin: [
                6,
                {
                    PERSON: 1,
                    PUSH: 2,
                    OTHER: 3,
                    BACKOFF: 4,
                },
            ],
            connectionSequenceStep: [
                11,
                {
                    PUSH_OVERRIDES: 1,
                    PRIMARY: 2,
                    PUSH_FALLBACKS: 4,
                    HOST_FALLBACK: 5,
                    NO_DNS: 6,
                    SOFTLAYER: 7,
                    PRIMARY_HTTP: 8,
                    SOFTLAYER_HTTP: 9,
                    HOST_FALLBACK_HTTP: 10,
                    NO_DNS_HTTP: 11,
                },
            ],
            connectionT: [5, 'timer'],
            dnsResolutionMethod: [
                12,
                {
                    SYSTEM: 1,
                    GOOGLE: 2,
                    HARDCODED: 3,
                    NO_DNS: 4,
                },
            ],
            loginDnsResolver: [
                13,
                {
                    SYSTEM: 1,
                    GOOGLE: 2,
                    HARDCODED: 3,
                    NO_DNS: 4,
                },
            ],
            loginIpSource: [
                14,
                {
                    PUSH_OVERRIDES: 1,
                    G_WHATSAPP_NET: 2,
                    PUSH_FALLBACKS: 3,
                    G_FALLBACK_WHATSAPP_NET: 4,
                    HARDCODED_LIST: 5,
                    EX_WHATSAPP_NET: 6,
                },
            ],
            loginPort: [
                15,
                {
                    P5222: 1,
                    P443: 2,
                    P80: 3,
                    UNKNOWN: 4,
                },
            ],
            loginResult: [
                1,
                {
                    OK: 1,
                    ERROR_UNKNOWN: 2,
                    SERVER_ERROR: 3,
                    SERVER_GOAWAY: 4,
                    NETWORK_ERROR: 5,
                    ANDROID_KEYSTORE_ERROR: 6,
                    CERTIFICATE_ERROR: 7,
                },
            ],
            loginT: [3, 'timer'],
            longConnect: [4, 'boolean'],
            noiseProtocolVersion: [16, 'integer'],
            passive: [8, 'boolean'],
            pendingAcksCount: [17, 'integer'],
            retryCount: [2, 'integer'],
            sequenceStep: [7, 'integer'],
            serverErrorCode: [9, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebcPageLoad',
        id: 642,
        props: {
            webcAppcacheStatus: [
                29,
                {
                    UNCACHED: 0,
                    IDLE: 1,
                    CHECKING: 2,
                    DOWNLOADING: 3,
                    UPDATEREADY: 4,
                    OBSOLETE: 5,
                },
            ],
            webcCached: [30, 'boolean'],
            webcConnectEnd: [10, 'timer'],
            webcConnectStart: [9, 'timer'],
            webcDomComplete: [19, 'timer'],
            webcDomContentLoadedEventEnd: [18, 'timer'],
            webcDomContentLoadedEventStart: [17, 'timer'],
            webcDomInteractive: [16, 'timer'],
            webcDomLoading: [15, 'timer'],
            webcDomainLookupEnd: [8, 'timer'],
            webcDomainLookupStart: [7, 'timer'],
            webcExeDone: [23, 'timer'],
            webcExeStart: [22, 'timer'],
            webcFetchStart: [6, 'timer'],
            webcInitialMountT: [38, 'timer'],
            webcInitialNavMountT: [39, 'timer'],
            webcInitialPanel: [42, 'string'],
            webcInitialPanelMountStartT: [43, 'timer'],
            webcInitialPanelMountT: [40, 'timer'],
            webcInitialPanelRenderT: [46, 'timer'],
            webcJsLoadT: [37, 'timer'],
            webcLoadEventEnd: [21, 'timer'],
            webcLoadEventStart: [20, 'timer'],
            webcLoadInForeground: [53, 'boolean'],
            webcMainScriptEnd: [45, 'timer'],
            webcMainScriptStart: [44, 'timer'],
            webcNativeLoadT: [36, 'timer'],
            webcNavigation: [
                32,
                {
                    NAVIGATE_NEXT: 0,
                    RELOAD: 1,
                    BACK_FORWARD: 2,
                    UNDEFINED: 255,
                },
            ],
            webcPageLoadT: [34, 'timer'],
            webcParallellyFetched: [41, 'boolean'],
            webcQrCode: [31, 'boolean'],
            webcRedirectCount: [33, 'number'],
            webcRedirectEnd: [5, 'timer'],
            webcRedirectStart: [4, 'timer'],
            webcRequestStart: [12, 'timer'],
            webcResponseEnd: [14, 'timer'],
            webcResponseStart: [13, 'timer'],
            webcSecureConnectionStart: [11, 'timer'],
            webcUnloadEventEnd: [3, 'timer'],
            webcUnloadEventStart: [2, 'timer'],
            webcWsAttempts: [28, 'number'],
            webcWsNormal: [27, 'timer'],
            webcWsOpening: [24, 'timer'],
            webcWsPairing: [25, 'timer'],
            webcWsSyncing: [26, 'timer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'OfflineResume',
        id: 3112,
        props: {
            affectedBySleepMode: [35, 'boolean'],
            chatThreadCount: [1, 'integer'],
            expectedOfflineCallCount: [23, 'integer'],
            expectedOfflineMessageCount: [17, 'integer'],
            expectedOfflineNotificationCount: [18, 'integer'],
            expectedOfflineReceiptCount: [19, 'integer'],
            isOfflineCompleteMissed: [2, 'boolean'],
            isResumeInForeground: [13, 'boolean'],
            isRunningFromServiceExtension: [22, 'boolean'],
            lastStanzaT: [3, 'timer'],
            mailboxAge: [14, 'integer'],
            mainScreenLoadT: [4, 'timer'],
            offlineCallCount: [24, 'integer'],
            offlineDecryptErrorCount: [5, 'integer'],
            offlineMessageCount: [6, 'integer'],
            offlineNotificationCount: [7, 'integer'],
            offlinePreviewT: [8, 'timer'],
            offlineProcessingT: [20, 'timer'],
            offlineReceiptCount: [9, 'integer'],
            offlineResumeResult: [
                21,
                {
                    COMPLETE: 1,
                    INCOMPLETE_UNKNOWN_ERROR: 2,
                    INCOMPLETE_DISCONNECT: 3,
                    INCOMPLETE_APP_RESTART: 4,
                },
            ],
            offlineSizeBytes: [10, 'integer'],
            onTrickleMode: [15, 'boolean'],
            pageLoadT: [11, 'timer'],
            passiveModeT: [25, 'timer'],
            preackCallCount: [26, 'integer'],
            preackMessageCount: [27, 'integer'],
            preackNotificationCount: [28, 'integer'],
            preackReceiptCount: [29, 'integer'],
            processedCallCount: [30, 'integer'],
            processedMessageCount: [31, 'integer'],
            processedNotificationCount: [32, 'integer'],
            processedReceiptCount: [33, 'integer'],
            socketConnectT: [12, 'timer'],
            transientOfflineSessionId: [34, 'string'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebcDbOpen',
        id: 1940,
        props: {
            webcDbName: [1, 'string'],
            webcDbOpenNumAttempts: [3, 'integer'],
            webcDbOpenWasSuccess: [2, 'boolean'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'PsIdUpdate',
        id: 2862,
        props: {
            isFromWamsys: [4, 'boolean'],
            psIdAction: [
                2,
                {
                    CREATED: 1,
                    ROTATED: 2,
                    DELETED: 3,
                },
            ],
            psIdKey: [1, 'integer'],
            psIdRotationFrequence: [3, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebDbVersionsSource',
        id: 4784,
        props: {
            webDbVersionSource: [
                1,
                {
                    KNOB: 1,
                    LOCAL: 2,
                    STATIC: 3,
                    KNOB_WITH_LOCAL_OVERRIDE: 4,
                },
            ],
            webSchemaInitiator: [
                2,
                {
                    MAIN: 1,
                    WEB_WORKER: 2,
                    SERVICE_WORKER: 3,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebcAssetLoad',
        id: 1358,
        props: {
            webcAssetCacheType: [
                4,
                {
                    UNCACHED: 0,
                    IDB: 1,
                    SW: 2,
                },
            ],
            webcAssetFromCache: [2, 'boolean'],
            webcAssetLoadT: [3, 'timer'],
            webcAssetName: [1, 'string'],
            webcAssetSize: [5, 'number'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebcStorageStat',
        id: 1504,
        props: {
            webcAgeOfStorage: [3, 'integer'],
            webcPackingEnabled: [4, 'boolean'],
            webcStorageQuota: [2, 'integer'],
            webcStorageUsage: [1, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'OfflineResumeStage',
        id: 3536,
        props: {
            attemptId: [13, 'integer'],
            chatThreadCount: [4, 'integer'],
            currentOfflineStage: [
                1,
                {
                    PAGE_LOAD: 1,
                    SOCKET_CONNECT: 2,
                    PROCESS_COMPLETE: 3,
                    SCREEN_LOAD: 4,
                    OFFLINE_PREVIEW: 5,
                    OFFLINE_COMPLETE_RECEIVED: 6,
                    PREACKS_SENT: 7,
                },
            ],
            isResumeInForeground: [5, 'boolean'],
            isResumeStartedInForeground: [14, 'boolean'],
            lastPushTimestampMs: [12, 'integer'],
            mailboxAge: [6, 'integer'],
            offlineCallCount: [15, 'integer'],
            offlineDecryptErrorCount: [7, 'integer'],
            offlineMessageCount: [8, 'integer'],
            offlineNotificationCount: [9, 'integer'],
            offlineReceiptCount: [10, 'integer'],
            offlineResumeMode: [
                11,
                {
                    RESUME_FROM_RESTART: 1,
                    RESUME_FROM_OPEN_TAB: 2,
                    UNKNOWN: 3,
                    CONNECT_REASON_USER: 4,
                    CONNECT_REASON_PUSH: 5,
                    CONNECT_REASON_BACKOFF: 6,
                },
            ],
            offlineSessionId: [2, 'string'],
            offlineSizeBytes: [16, 'integer'],
            offlineStageTimestampMs: [3, 'integer'],
            passiveModeT: [17, 'timer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebcStreamModeChange',
        id: 770,
        props: {
            webcStreamMode: [
                1,
                {
                    QR: 0,
                    MAIN: 1,
                    SYNCING: 2,
                    OFFLINE: 3,
                    CONFLICT: 4,
                    PROXYBLOCK: 5,
                    TOS_BLOCK: 6,
                    SMB_TOS_BLOCK: 7,
                    DEPRECATED_VERSION: 8,
                    LOCK: 9,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebcStatusSync',
        id: 1878,
        props: {
            webcStatusMutedItemCount: [4, 'integer'],
            webcStatusMutedRowCount: [7, 'integer'],
            webcStatusRecentItemCount: [2, 'integer'],
            webcStatusRecentRowCount: [5, 'integer'],
            webcStatusSyncT: [1, 'timer'],
            webcStatusViewedItemCount: [3, 'integer'],
            webcStatusViewedRowCount: [6, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'TsNavigation',
        id: 4334,
        props: {
            cid: [19, 'string'],
            entryPointConversationInitiated: [
                23,
                {
                    BUSINESS_INITIATED: 0,
                    CONSUMER_INITIATED: 1,
                    NO_MESSAGES_LAST_24H: 2,
                },
            ],
            entryPointConversionApp: [24, 'string'],
            entryPointConversionSource: [25, 'string'],
            groupSize: [1, 'integer'],
            isCatalogVisible: [29, 'boolean'],
            isCloudapi: [26, 'boolean'],
            isOnpremises: [27, 'boolean'],
            isSmb: [28, 'boolean'],
            navigationDestination: [
                2,
                {
                    UNKNOWN: 1,
                    COMMUNITY_HOME: 2,
                    COMMUNITY_TAB: 3,
                    CHAT_THREAD: 4,
                    CHAT_INFO_PAGE: 5,
                    CHAT_LIST: 6,
                    COMMUNITY_NAVIGATION: 7,
                    BACKGROUND: 8,
                    RESIGN: 9,
                    SUBGROUP_SWITCHER: 10,
                    COMMUNITY_COMPOSER: 11,
                    MEDIA_VIEWER: 12,
                    ALL_MEDIA: 13,
                    STREAMING_MEDIA_VIEWER: 14,
                    CALLS_HOME_TAB: 15,
                    IN_CALL_UI: 16,
                    STATUS_HOME_TAB: 17,
                    STATUS_COMPOSER: 18,
                    STATUS_VIEW: 19,
                    CAMERA: 20,
                    PHOTO_VIDEO_LIBRARY: 21,
                    SETTINGS_HOME_TAB: 22,
                    TOOLS_TAB: 23,
                    COMMUNITY_SETTINGS: 24,
                    CHANNEL_UPDATES_HOME: 25,
                    CHANNEL_THREAD: 26,
                    CHANNEL_DIRECTORY: 27,
                    CHANNEL_PROFILE: 28,
                    CHANNEL_FORWARD: 29,
                    CHANNEL_PRODUCER_MEDIA_EDITOR: 30,
                    CHANNEL_CREATE: 31,
                    CHANNEL_EDIT: 32,
                    CHANNEL_DELETE: 33,
                    STATUS_COMPOSER_TEXT: 34,
                    LIGHTWEIGHT_CALL_UI: 35,
                    AI_HOME: 36,
                    LIGHTWEIGHT_CALL_BANNER: 37,
                    LIGHTWEIGHT_CALL_NUX: 38,
                    CALL_INFO_PAGE: 39,
                    SMB_CATALOG: 40,
                    SMB_CATALOG_SHARE_LINK: 41,
                    SMB_CATALOG_SHARE_PRODUCT_LINK: 42,
                    SMB_CATALOG_SETTINGS: 43,
                    SMB_CATALOG_PRODUCT: 44,
                    SMB_CATALOG_NEW_COLLECTION: 45,
                    SMB_CATALOG_EDIT_COLLECTION: 46,
                    SMB_CATALOG_CREATE_PRODUCT: 47,
                    SMB_CATALOG_EDIT_PRODUCT: 48,
                    SMB_CATALOG_NEW_COLLECTION_ADD_ITEMS: 49,
                    SMB_CATALOG_COLLECTION_UPDATE_ITEMS: 50,
                    SMB_CATALOG_COLLECTIONS: 51,
                    SMB_CATALOG_COLLECTION_REORDER: 52,
                    SMB_CATALOG_COLLECTION_PRODUCTS: 53,
                    SMB_CATALOG_COLLECTION_PRODUCTS_REORDER: 54,
                    CREATE_EVENT: 55,
                    EVENT_PAGE: 56,
                    EVENTS_IN_GROUP: 57,
                    CATALOG_HOME: 59,
                    CATALOG_COLLECTION: 60,
                    CATALOG_ITEM: 61,
                    CATALOG_CART: 62,
                    FLOWS_EXPERIENCE: 63,
                    BUSINESS_PROFILE_EDIT: 64,
                    BUSINESS_PROFILE_EDIT_NAME: 65,
                    BUSINESS_PROFILE_EDIT_CATEGORIES: 66,
                    BUSINESS_PROFILE_EDIT_DESCRIPTION: 67,
                    BUSINESS_PROFILE_EDIT_WEBSITE: 68,
                    BUSINESS_PROFILE_EDIT_EMAIL: 69,
                    BUSINESS_PROFILE_EDIT_ADDRESS: 70,
                    BUSINESS_PROFILE_EDIT_HOURS: 71,
                    BUSINESS_PROFILE_EDIT_COVER_PHOTO: 72,
                    BUSINESS_PROFILE_LINK_FB_IG: 73,
                    BUSINESS_PROFILE_HELP_PAGE_LINK_FB_IG: 74,
                    PAYMENTS_HOME: 75,
                    PAYMENTS_HOME_TRANSACTIONS: 76,
                    PAYMENTS_HOME_RECENT_MERCHANTS: 77,
                    PAYMENTS_HOME_INVITE_OTHERS: 78,
                    PAYMENTS_HOME_HELP: 79,
                    MESSAGE_MENU: 80,
                    PIN_MESSAGE_REPLACEMENT_DIALOG: 81,
                    PIN_MESSAGE_EXPIRATION_DIALOG: 82,
                    MY_STATUS_LIST: 83,
                    ALL_STATUS_LIST: 84,
                    PIN_BANNER: 85,
                    CALL_BANNER: 86,
                    LANDING_SCREEN: 87,
                    NEW_CHAT: 88,
                    GROUP_MEMBER_ADD_GROUP_CREATION: 89,
                    GROUP_MEMBER_ADD_EXISTING_GROUP: 90,
                    GROUP_CHAT: 91,
                    GROUP_CREATION: 92,
                },
            ],
            navigationSource: [
                3,
                {
                    UNKNOWN: 1,
                    COMMUNITY_HOME: 2,
                    COMMUNITY_TAB: 3,
                    CHAT_THREAD: 4,
                    CHAT_INFO_PAGE: 5,
                    CHAT_LIST: 6,
                    COMMUNITY_NAVIGATION: 7,
                    BACKGROUND: 8,
                    RESIGN: 9,
                    SUBGROUP_SWITCHER: 10,
                    COMMUNITY_COMPOSER: 11,
                    MEDIA_VIEWER: 12,
                    ALL_MEDIA: 13,
                    STREAMING_MEDIA_VIEWER: 14,
                    CALLS_HOME_TAB: 15,
                    IN_CALL_UI: 16,
                    STATUS_HOME_TAB: 17,
                    STATUS_COMPOSER: 18,
                    STATUS_VIEW: 19,
                    CAMERA: 20,
                    PHOTO_VIDEO_LIBRARY: 21,
                    SETTINGS_HOME_TAB: 22,
                    TOOLS_TAB: 23,
                    COMMUNITY_SETTINGS: 24,
                    CHANNEL_UPDATES_HOME: 25,
                    CHANNEL_THREAD: 26,
                    CHANNEL_DIRECTORY: 27,
                    CHANNEL_PROFILE: 28,
                    CHANNEL_FORWARD: 29,
                    CHANNEL_PRODUCER_MEDIA_EDITOR: 30,
                    CHANNEL_CREATE: 31,
                    CHANNEL_EDIT: 32,
                    CHANNEL_DELETE: 33,
                    STATUS_COMPOSER_TEXT: 34,
                    LIGHTWEIGHT_CALL_UI: 35,
                    AI_HOME: 36,
                    LIGHTWEIGHT_CALL_BANNER: 37,
                    LIGHTWEIGHT_CALL_NUX: 38,
                    CALL_INFO_PAGE: 39,
                    SMB_CATALOG: 40,
                    SMB_CATALOG_SHARE_LINK: 41,
                    SMB_CATALOG_SHARE_PRODUCT_LINK: 42,
                    SMB_CATALOG_SETTINGS: 43,
                    SMB_CATALOG_PRODUCT: 44,
                    SMB_CATALOG_NEW_COLLECTION: 45,
                    SMB_CATALOG_EDIT_COLLECTION: 46,
                    SMB_CATALOG_CREATE_PRODUCT: 47,
                    SMB_CATALOG_EDIT_PRODUCT: 48,
                    SMB_CATALOG_NEW_COLLECTION_ADD_ITEMS: 49,
                    SMB_CATALOG_COLLECTION_UPDATE_ITEMS: 50,
                    SMB_CATALOG_COLLECTIONS: 51,
                    SMB_CATALOG_COLLECTION_REORDER: 52,
                    SMB_CATALOG_COLLECTION_PRODUCTS: 53,
                    SMB_CATALOG_COLLECTION_PRODUCTS_REORDER: 54,
                    CREATE_EVENT: 55,
                    EVENT_PAGE: 56,
                    EVENTS_IN_GROUP: 57,
                    CATALOG_HOME: 59,
                    CATALOG_COLLECTION: 60,
                    CATALOG_ITEM: 61,
                    CATALOG_CART: 62,
                    FLOWS_EXPERIENCE: 63,
                    BUSINESS_PROFILE_EDIT: 64,
                    BUSINESS_PROFILE_EDIT_NAME: 65,
                    BUSINESS_PROFILE_EDIT_CATEGORIES: 66,
                    BUSINESS_PROFILE_EDIT_DESCRIPTION: 67,
                    BUSINESS_PROFILE_EDIT_WEBSITE: 68,
                    BUSINESS_PROFILE_EDIT_EMAIL: 69,
                    BUSINESS_PROFILE_EDIT_ADDRESS: 70,
                    BUSINESS_PROFILE_EDIT_HOURS: 71,
                    BUSINESS_PROFILE_EDIT_COVER_PHOTO: 72,
                    BUSINESS_PROFILE_LINK_FB_IG: 73,
                    BUSINESS_PROFILE_HELP_PAGE_LINK_FB_IG: 74,
                    PAYMENTS_HOME: 75,
                    PAYMENTS_HOME_TRANSACTIONS: 76,
                    PAYMENTS_HOME_RECENT_MERCHANTS: 77,
                    PAYMENTS_HOME_INVITE_OTHERS: 78,
                    PAYMENTS_HOME_HELP: 79,
                    MESSAGE_MENU: 80,
                    PIN_MESSAGE_REPLACEMENT_DIALOG: 81,
                    PIN_MESSAGE_EXPIRATION_DIALOG: 82,
                    MY_STATUS_LIST: 83,
                    ALL_STATUS_LIST: 84,
                    PIN_BANNER: 85,
                    CALL_BANNER: 86,
                    LANDING_SCREEN: 87,
                    NEW_CHAT: 88,
                    GROUP_MEMBER_ADD_GROUP_CREATION: 89,
                    GROUP_MEMBER_ADD_EXISTING_GROUP: 90,
                    GROUP_CHAT: 91,
                    GROUP_CREATION: 92,
                },
            ],
            relativeTimestampMs: [4, 'integer'],
            smbCatalogBusinessVertical: [20, 'string'],
            smbCatalogIsCatalogVisible: [21, 'boolean'],
            smbCatalogIsToggleCart: [22, 'boolean'],
            threadType: [
                7,
                {
                    GROUP: 1,
                    BROADCAST: 2,
                    INDIVIDUAL: 3,
                    STATUS: 4,
                    CHANNEL: 5,
                    SUB_GROUP: 6,
                    DEFAULT_SUB_GROUP: 7,
                    PARENT_GROUP: 8,
                },
            ],
            tsSessionId: [5, 'integer'],
            typeOfGroup: [
                6,
                {
                    GROUP: 1,
                    SUBGROUP: 2,
                    DEFAULT_SUBGROUP: 3,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'UserActivity',
        id: 1384,
        props: {
            userActivityBitmapHigh: [4, 'integer'],
            userActivityBitmapLen: [5, 'integer'],
            userActivityBitmapLow: [3, 'integer'],
            userActivitySessionCum: [7, 'integer'],
            userActivitySessionId: [1, 'string'],
            userActivitySessionSeq: [6, 'integer'],
            userActivityStartTime: [2, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'TsBitArray',
        id: 4332,
        props: {
            bitarrayHigh: [1, 'integer'],
            bitarrayLength: [2, 'integer'],
            bitarrayLow: [3, 'integer'],
            cumulativeBits: [4, 'integer'],
            relativeTimestampMs: [5, 'integer'],
            sessionSeq: [6, 'integer'],
            tsSessionId: [7, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebcResourceLoad',
        id: 688,
        props: {
            webcResourceCached: [3, 'boolean'],
            webcResourceDuration: [2, 'timer'],
            webcResourceName: [1, 'string'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MdAppStateMessageRange',
        id: 2522,
        props: {
            additionalMessagesCount: [1, 'integer'],
        },
        weight: 1000,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MdBootstrapDataApplied',
        id: 2298,
        props: {
            collection: [
                5,
                {
                    REGULAR: 1,
                    REGULAR_LOW: 2,
                    REGULAR_HIGH: 3,
                    CRITICAL_BLOCK: 4,
                    CRITICAL_UNBLOCK_LOW: 5,
                },
            ],
            historySyncChunkOrder: [14, 'integer'],
            historySyncStageProgress: [11, 'integer'],
            mdBootstrapContactsCount: [8, 'integer'],
            mdBootstrapHistoryPayloadType: [
                10,
                {
                    INITIAL: 1,
                    RECENT_HISTORY: 2,
                    FULL_HISTORY: 3,
                    PUSHNAME: 4,
                    STATUS_V3: 5,
                    NON_BLOCKING_DATA: 6,
                    ON_DEMAND: 7,
                },
            ],
            mdBootstrapPayloadType: [
                3,
                {
                    CRITICAL: 1,
                    NON_CRITICAL: 2,
                },
            ],
            mdBootstrapSource: [
                2,
                {
                    APP_STATE: 1,
                    HISTORY: 2,
                },
            ],
            mdBootstrapStepDuration: [6, 'integer'],
            mdBootstrapStepResult: [
                12,
                {
                    SUCCESS: 1,
                    FAILURE: 2,
                },
            ],
            mdRegAttemptId: [9, 'string'],
            mdSessionId: [1, 'string'],
            mdTimestamp: [4, 'integer'],
            sentViaMms: [13, 'boolean'],
            usedSnapshot: [7, 'boolean'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MdBootstrapAppStateDataDownloaded',
        id: 2294,
        props: {
            mdBootstrapContactsCount: [5, 'integer'],
            mdBootstrapHistoryPayloadType: [
                11,
                {
                    INITIAL: 1,
                    RECENT_HISTORY: 2,
                    FULL_HISTORY: 3,
                    PUSHNAME: 4,
                    STATUS_V3: 5,
                    NON_BLOCKING_DATA: 6,
                    ON_DEMAND: 7,
                },
            ],
            mdBootstrapPayloadSize: [4, 'integer'],
            mdBootstrapPayloadType: [
                2,
                {
                    CRITICAL: 1,
                    NON_CRITICAL: 2,
                },
            ],
            mdBootstrapStepDuration: [6, 'integer'],
            mdBootstrapStepResult: [
                7,
                {
                    SUCCESS: 1,
                    FAILURE: 2,
                },
            ],
            mdRegAttemptId: [10, 'string'],
            mdSessionId: [1, 'string'],
            mdStorageQuotaBytes: [8, 'integer'],
            mdStorageQuotaUsedBytes: [9, 'integer'],
            mdTimestamp: [3, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MdAppStateKeyRotation',
        id: 2518,
        props: {
            mdAppStateKeyRotationReason: [
                1,
                {
                    APP_STATE_SYNC_KEY_EXPIRY: 1,
                    DEVICE_DEREGISTERATION: 2,
                    NO_KEYS: 3,
                },
            ],
        },
        weight: 1000,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MdAppStateSyncDaily',
        id: 2300,
        props: {
            crossIndexConflictCount: [11, 'integer'],
            invalidActionCount: [4, 'integer'],
            keyRotationRemoveCount: [12, 'integer'],
            missingKeyCount: [9, 'integer'],
            mutationCount: [1, 'integer'],
            storedMutationCount: [7, 'integer'],
            unsetActionCount: [8, 'integer'],
            unsupportedActionCount: [5, 'integer'],
            uploadConflictCount: [10, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebcPwaEvent',
        id: 4116,
        props: {
            webcPwaAction: [
                2,
                {
                    INSTALL: 1,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'DeepLinkClick',
        id: 1156,
        props: {
            deepLinkHasPhoneNumber: [2, 'boolean'],
            deepLinkHasText: [1, 'boolean'],
            deepLinkSessionId: [3, 'string'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MessageSecretErrors',
        id: 3686,
        props: {
            messageMediaType: [
                3,
                {
                    NONE: 1,
                    PHOTO: 2,
                    VIDEO: 3,
                    AUDIO: 4,
                    PTT: 5,
                    LOCATION: 6,
                    CONTACT: 7,
                    DOCUMENT: 8,
                    URL: 9,
                    CALL: 10,
                    GIF: 11,
                    FUTURE: 12,
                    CONTACT_ARRAY: 13,
                    LIVE_LOCATION: 14,
                    PROFILE_PIC: 15,
                    STICKER: 16,
                    HSM: 17,
                    PRODUCT_IMAGE: 18,
                    TEMPLATE: 19,
                    MD_APP_STATE: 20,
                    MD_HISTORY_SYNC: 21,
                    CATALOG_LINK: 22,
                    PRODUCT_LINK: 23,
                    ORDER: 24,
                    LIST: 25,
                    LIST_REPLY: 26,
                    BUTTON_MESSAGE: 27,
                    BUTTON_RESPONSE_MESSAGE: 28,
                    PRODUCT_LIST: 29,
                    BUTTON_NFM: 30,
                    VIEW_ONCE_IMAGE: 31,
                    VIEW_ONCE_VIDEO: 32,
                    SHOP_STOREFRONT: 33,
                    REACTION: 34,
                    INTERACTIVE_NFM: 35,
                    INTERACTIVE_RESPONSE_NFM: 36,
                    POLL_CREATE: 37,
                    POLL_VOTE: 38,
                    TEXT: 39,
                    KEEP: 40,
                    UNKEEP: 41,
                    INVISIBLE_HELLO: 42,
                    MEDIA_EXPRESS_NOTIFY: 43,
                    UNDECRYPTED_MEDIA: 44,
                    UNDECRYPTED_TEXT: 45,
                    UNDECRYPTED_PAY: 46,
                    UNDECRYPTED_POLL: 47,
                    UNDECRYPTED_REACTION: 48,
                    UNDECRYPTED_MEDIA_NOTIFY: 49,
                    VIEW_ONCE_AUDIO: 50,
                    SCHEDULED_CALL_CREATE: 51,
                    SCHEDULED_CALL_CANCEL: 52,
                    PUSH_TO_VIDEO: 53,
                    PIN_IN_CHAT: 54,
                    INTERACTIVE_CAROUSEL: 55,
                    COMMENT: 56,
                    EPHEMERAL_SYNC_RESPONSE: 57,
                    EVENT_CREATE: 58,
                    EVENT_RESPOND: 59,
                    LOTTIE_STICKER: 60,
                    INTERACTIVE_PRODUCT_CAROUSEL: 61,
                    INTERACTIVE_PRODUCT: 62,
                },
            ],
            messageSecretAllowedList: [
                1,
                {
                    MESSAGE_POLL: 0,
                },
            ],
            messageSecretError: [
                2,
                {
                    MISSING_MESSAGE_SECRET: 0,
                    WRONG_LENGTH: 1,
                    ENCRYPTION_ERROR: 2,
                    DECRYPTION_ERROR: 3,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'ChatMessageCounts',
        id: 1644,
        props: {
            audioMessagesReceived: [149, 'integer'],
            audioMessagesSent: [150, 'integer'],
            awayMsgsSent: [56, 'integer'],
            bizCatalogType: [
                60,
                {
                    SHOPS: 0,
                    NATIVE: 1,
                },
            ],
            bizConversationDepth: [65, 'integer'],
            blockReason: [
                33,
                {
                    OTHER: 0,
                    NO_LONGER_NEEDED: 1,
                    NO_SIGN_UP: 2,
                    SPAM: 3,
                    OFFENSIVE_MESSAGES: 4,
                    OTP_DID_NOT_REQUEST: 5,
                },
            ],
            broadcastMsgsReceived: [30, 'integer'],
            broadcastMsgsSent: [29, 'integer'],
            callOffersReceived: [27, 'integer'],
            callOffersSent: [26, 'integer'],
            callsResultBusy: [70, 'integer'],
            callsResultCancelled: [71, 'integer'],
            callsResultConnected: [72, 'integer'],
            callsResultError: [78, 'integer'],
            callsResultMissed: [73, 'integer'],
            callsResultRejected: [74, 'integer'],
            canEditDmSettings: [88, 'boolean'],
            cartViews: [86, 'integer'],
            chatEphemeralityDuration: [15, 'integer'],
            chatMuted: [
                8,
                {
                    NOT_MUTED: 1,
                    MUTED_NO_NOTIFICATIONS: 2,
                    MUTED_SILENT_NOTIFICATIONS: 3,
                },
            ],
            chatOrigins: [
                179,
                {
                    LID_USERNAME: 1,
                    LID_CTWA: 2,
                    OTHERS: 3,
                },
            ],
            chatOverflowClicks: [79, 'integer'],
            chatTypeInd: [
                2,
                {
                    INDIVIDUAL: 1,
                    SMB: 2,
                    ENT: 3,
                    INTEROP: 4,
                },
            ],
            collectionInquiriesSent: [44, 'integer'],
            commandSheetShow: [174, 'integer'],
            commentsReceived: [126, 'integer'],
            commerceMsgsReceived: [41, 'integer'],
            commerceMsgsSent: [40, 'integer'],
            disappearingChatInitiator: [
                59,
                {
                    CHAT: 1,
                    INITIATED_BY_ME: 2,
                    INITIATED_BY_OTHER: 3,
                    CHAT_PICKER: 4,
                    BIZ_UPGRADE_FB_HOSTING: 5,
                },
            ],
            documentMessagesReceived: [151, 'integer'],
            documentMessagesSent: [152, 'integer'],
            editedMsgsSent: [100, 'integer'],
            entryPointConversionApp: [47, 'string'],
            entryPointConversionSource: [46, 'string'],
            ephemeralMessagesReceived: [14, 'integer'],
            ephemeralMessagesSent: [13, 'integer'],
            ephemeralMessagesUnreadExpired: [69, 'integer'],
            ephemeralityInitiator: [
                109,
                {
                    INITIATED_BY_ME: 1,
                    INITIATED_BY_OTHER: 2,
                    BIZ_UPGRADE_FB_HOSTING: 3,
                },
            ],
            ephemeralityTriggerAction: [
                110,
                {
                    UNKNOWN: 0,
                    CHAT_SETTINGS: 1,
                    ACCOUNT_SETTINGS: 2,
                    BULK_CHANGE: 3,
                    BIZ_SUPPORTS_FB_HOSTING: 4,
                },
            ],
            eventCreationMessagesReceived: [142, 'integer'],
            eventCreationMessagesSent: [143, 'integer'],
            eventResponseMessagesReceived: [144, 'integer'],
            eventResponseMessagesSent: [145, 'integer'],
            fbCtaInquiriesSent: [45, 'integer'],
            firstResponseTime: [25, 'integer'],
            forwardAudioMessagesReceived: [153, 'integer'],
            forwardAudioMessagesSent: [154, 'integer'],
            forwardDocumentMessagesReceived: [155, 'integer'],
            forwardDocumentMessagesSent: [156, 'integer'],
            forwardGifMessagesReceived: [157, 'integer'],
            forwardGifMessagesSent: [158, 'integer'],
            forwardMessagesReceived: [89, 'integer'],
            forwardMessagesSent: [90, 'integer'],
            forwardPhotoMessagesReceived: [128, 'integer'],
            forwardPhotoMessagesSent: [129, 'integer'],
            forwardPtvMessagesReceived: [159, 'integer'],
            forwardPtvMessagesSent: [160, 'integer'],
            forwardStatusReplyMessagesReceived: [161, 'integer'],
            forwardStatusReplyMessagesSent: [162, 'integer'],
            forwardStickerMessagesReceived: [163, 'integer'],
            forwardStickerMessagesSent: [164, 'integer'],
            forwardTextMessagesReceived: [130, 'integer'],
            forwardTextMessagesSent: [131, 'integer'],
            forwardUrlMessagesReceived: [132, 'integer'],
            forwardUrlMessagesSent: [133, 'integer'],
            forwardVideoMessagesReceived: [165, 'integer'],
            forwardVideoMessagesSent: [166, 'integer'],
            gaStatus: [
                22,
                {
                    NEW: 0,
                    RETAINED: 1,
                    RESURRECTED: 2,
                },
            ],
            gifMessagesReceived: [167, 'integer'],
            gifMessagesSent: [168, 'integer'],
            greetingMsgsSent: [57, 'integer'],
            groupContainsBiz: [75, 'boolean'],
            groupInfoSetting: [
                87,
                {
                    ADMINS_ONLY: 1,
                    ALL_PARTICIPANTS: 2,
                },
            ],
            groupMembershipReplies: [51, 'integer'],
            groupPrivateReplies: [52, 'integer'],
            groupSize: [19, 'integer'],
            hasUsername: [180, 'boolean'],
            imagineCommandClick: [175, 'integer'],
            imagineMentionClick: [146, 'integer'],
            imagineMentionShow: [176, 'integer'],
            isAContact: [6, 'boolean'],
            isAGroup: [5, 'boolean'],
            isArchived: [10, 'boolean'],
            isAutoMuted: [108, 'boolean'],
            isBlocked: [32, 'boolean'],
            isCartAddClicked: [36, 'boolean'],
            isCommerceViewed: [35, 'boolean'],
            isCtaOnPdpClicked: [37, 'boolean'],
            isInviteCreatedThread: [106, 'boolean'],
            isLabelled: [54, 'boolean'],
            isMessageYourself: [91, 'boolean'],
            isOppositePartyInitiated: [62, 'boolean'],
            isPinned: [9, 'boolean'],
            isPnhEnabledChat: [92, 'boolean'],
            isUser1pBizBotChat: [141, 'boolean'],
            isUser3pBotChat: [140, 'boolean'],
            isUserAgent: [111, 'boolean'],
            isUserCreatedAgent: [123, 'boolean'],
            isWaPayRegistered: [81, 'boolean'],
            labelledMsgs: [55, 'integer'],
            locationsSent: [96, 'integer'],
            messagesRead: [127, 'integer'],
            messagesReceived: [4, 'integer'],
            messagesSent: [3, 'integer'],
            messagesStarred: [12, 'integer'],
            messagesUnread: [11, 'integer'],
            metaAiMentionClick: [147, 'integer'],
            metaAiMentionShow: [148, 'integer'],
            newThread: [68, 'boolean'],
            odReceived: [82, 'integer'],
            oppositeVisibleIdentification: [
                181,
                {
                    PHONE_NUMBER: 1,
                    SAVED_CONTACT_NAME: 2,
                    USERNAME: 3,
                    MASKED_PHONE_NUMBER: 4,
                    VERIFIED_BUSINESS_NAME: 5,
                    PLACEHOLDER: 6,
                    PUSHNAME: 7,
                },
            ],
            ordersSent: [38, 'integer'],
            p2mOdNnpTransactionsSent: [83, 'integer'],
            p2mOdNpTransactionsSent: [84, 'integer'],
            paymentsSent: [39, 'integer'],
            pdpInquiriesSent: [42, 'integer'],
            pdpViews: [61, 'integer'],
            photoMessagesReceived: [134, 'integer'],
            photoMessagesSent: [135, 'integer'],
            pollCreationMessagesReceived: [115, 'integer'],
            pollCreationMessagesSent: [116, 'integer'],
            pollUpdateMessagesReceived: [117, 'integer'],
            pollUpdateMessagesSent: [118, 'integer'],
            profileReplies: [64, 'integer'],
            profileViews: [63, 'integer'],
            pttMessagesReceived: [119, 'integer'],
            pttMessagesSent: [120, 'integer'],
            ptvMessagesReceived: [121, 'integer'],
            ptvMessagesSent: [122, 'integer'],
            quickRepliesSent: [58, 'integer'],
            reactionsReceived: [97, 'integer'],
            reactionsSent: [98, 'integer'],
            receiverDefaultDisappearingDuration: [21, 'integer'],
            repliesSent: [80, 'integer'],
            requestedPhoneNumber: [93, 'boolean'],
            seenMaskedPhoneNumber: [94, 'boolean'],
            senderDefaultDisappearingDuration: [20, 'integer'],
            sharedPhoneNumber: [99, 'boolean'],
            smbMarketingMessagesReactionsSent: [101, 'integer'],
            smbMarketingMessagesRepliesSent: [102, 'integer'],
            smbMarketingMessagesSpamReports: [103, 'integer'],
            smbMarketingMsgsReceived: [104, 'integer'],
            smbMarketingMsgsSent: [105, 'integer'],
            spamReports: [31, 'integer'],
            startTime: [7, 'integer'],
            statusReplies: [50, 'integer'],
            statusReplyMessagesReceived: [169, 'integer'],
            statusViews: [49, 'integer'],
            stickerMessagesReceived: [170, 'integer'],
            stickerMessagesSent: [171, 'integer'],
            storefrontInquiriesSent: [43, 'integer'],
            suggestionPromptsClick: [177, 'integer'],
            suggestionPromptsShow: [178, 'integer'],
            textMessagesReceived: [136, 'integer'],
            textMessagesSent: [137, 'integer'],
            textMessagesToUserCreatedAgentCnt: [124, 'integer'],
            threadDs: [66, 'string'],
            threadId: [67, 'string'],
            totalCallDuration: [28, 'integer'],
            totalMessageEditsFromAgentCnt: [112, 'integer'],
            totalMessageFromAgentCnt: [113, 'integer'],
            totalMessageToAgentCnt: [114, 'integer'],
            totalMessagesToUserCreatedAgentCnt: [125, 'integer'],
            typeOfGroup: [
                85,
                {
                    GROUP: 1,
                    SUBGROUP: 2,
                    DEFAULT_SUBGROUP: 3,
                },
            ],
            urlMessagesReceived: [138, 'integer'],
            urlMessagesSent: [139, 'integer'],
            viaContactlessChats: [107, 'boolean'],
            videoCallsOffered: [76, 'integer'],
            videoMessagesReceived: [172, 'integer'],
            videoMessagesSent: [173, 'integer'],
            viewOnceMessagesOpened: [18, 'integer'],
            viewOnceMessagesReceived: [17, 'integer'],
            viewOnceMessagesSent: [16, 'integer'],
            voiceCallsOffered: [77, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'AddressingModeMismatch',
        id: 4750,
        props: {
            iqResponse: [
                1,
                {
                    ADD_PARTICIPANT: 1,
                    REMOVE_PARTICIPANT: 2,
                    PROMOTE_PARTICIPANT: 3,
                    DEMOTE_PARTICIPANT: 4,
                },
            ],
            localAddressingMode: [
                2,
                {
                    PN: 1,
                    LID: 2,
                },
            ],
            mismatchOrigin: [
                6,
                {
                    INCOMING_GROUP_MESSAGE: 1,
                    ACK_OUTGOING_MESSAGE: 2,
                    GROUP_NOTIFICATION: 3,
                    GROUP_PROFILE_PICTURE_NOTIFICATION: 4,
                    IQ_RESPONSES: 5,
                },
            ],
            notificationTag: [3, 'string'],
            serverAddressingMode: [
                5,
                {
                    PN: 1,
                    LID: 2,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MdBadDeviceSentMessage',
        id: 2176,
        props: {
            dsmError: [
                2,
                {
                    INVALID_SENDER: 1,
                    MISSING_DSM: 2,
                    INVALID_DSM: 3,
                },
            ],
            peerType: [
                1,
                {
                    PRIMARY: 1,
                    COMPANION: 2,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'GatedMessageReceived',
        id: 3152,
        props: {
            chatGatedReason: [
                1,
                {
                    TOS3: 1,
                    COUNTRY: 2,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MessageReceive',
        id: 450,
        props: {
            agentEngagementType: [
                28,
                {
                    DIRECT_CHAT: 0,
                    INVOKED: 1,
                    MEMBER: 2,
                },
            ],
            botType: [
                37,
                {
                    UNKNOWN: 0,
                    METABOT: 1,
                    BOT_1P_BIZ: 2,
                    BOT_3P_BIZ: 3,
                },
            ],
            chatOrigins: [
                38,
                {
                    LID_USERNAME: 1,
                    LID_CTWA: 2,
                    OTHERS: 3,
                },
            ],
            deviceCount: [16, 'integer'],
            deviceSizeBucket: [
                10,
                {
                    LT32: 1,
                    LT64: 2,
                    LT128: 3,
                    LT256: 4,
                    LT512: 5,
                    LT1000: 6,
                    LT1024: 16,
                    LT1500: 7,
                    LT2000: 8,
                    LT2500: 9,
                    LT3000: 10,
                    LT3500: 11,
                    LT4000: 12,
                    LT4500: 13,
                    LT5000: 14,
                    LARGEST_BUCKET: 15,
                },
            ],
            disappearingChatInitiator: [
                14,
                {
                    CHAT: 1,
                    INITIATED_BY_ME: 2,
                    INITIATED_BY_OTHER: 3,
                    CHAT_PICKER: 4,
                    BIZ_UPGRADE_FB_HOSTING: 5,
                },
            ],
            editType: [
                25,
                {
                    NOT_EDITED: 0,
                    EDITED: 1,
                    SENDER_REVOKE: 2,
                    ADMIN_REVOKE: 3,
                },
            ],
            ephemeralityDuration: [13, 'integer'],
            ephemeralityInitiator: [
                26,
                {
                    INITIATED_BY_ME: 1,
                    INITIATED_BY_OTHER: 2,
                    BIZ_UPGRADE_FB_HOSTING: 3,
                },
            ],
            ephemeralityTriggerAction: [
                27,
                {
                    UNKNOWN: 0,
                    CHAT_SETTINGS: 1,
                    ACCOUNT_SETTINGS: 2,
                    BULK_CHANGE: 3,
                    BIZ_SUPPORTS_FB_HOSTING: 4,
                },
            ],
            hasUsername: [39, 'boolean'],
            isAComment: [36, 'boolean'],
            isAReply: [19, 'boolean'],
            isForwardedForward: [18, 'boolean'],
            isLid: [24, 'boolean'],
            isViewOnce: [9, 'boolean'],
            localAddressingMode: [
                33,
                {
                    PN: 1,
                    LID: 2,
                },
            ],
            messageAddressingMode: [
                34,
                {
                    PN: 1,
                    LID: 2,
                },
            ],
            messageIsInternational: [4, 'boolean'],
            messageIsInvisible: [23, 'boolean'],
            messageIsOffline: [5, 'boolean'],
            messageMediaType: [
                2,
                {
                    NONE: 1,
                    PHOTO: 2,
                    VIDEO: 3,
                    AUDIO: 4,
                    PTT: 5,
                    LOCATION: 6,
                    CONTACT: 7,
                    DOCUMENT: 8,
                    URL: 9,
                    CALL: 10,
                    GIF: 11,
                    FUTURE: 12,
                    CONTACT_ARRAY: 13,
                    LIVE_LOCATION: 14,
                    PROFILE_PIC: 15,
                    STICKER: 16,
                    HSM: 17,
                    PRODUCT_IMAGE: 18,
                    TEMPLATE: 19,
                    MD_APP_STATE: 20,
                    MD_HISTORY_SYNC: 21,
                    CATALOG_LINK: 22,
                    PRODUCT_LINK: 23,
                    ORDER: 24,
                    LIST: 25,
                    LIST_REPLY: 26,
                    BUTTON_MESSAGE: 27,
                    BUTTON_RESPONSE_MESSAGE: 28,
                    PRODUCT_LIST: 29,
                    BUTTON_NFM: 30,
                    VIEW_ONCE_IMAGE: 31,
                    VIEW_ONCE_VIDEO: 32,
                    SHOP_STOREFRONT: 33,
                    REACTION: 34,
                    INTERACTIVE_NFM: 35,
                    INTERACTIVE_RESPONSE_NFM: 36,
                    POLL_CREATE: 37,
                    POLL_VOTE: 38,
                    TEXT: 39,
                    KEEP: 40,
                    UNKEEP: 41,
                    INVISIBLE_HELLO: 42,
                    MEDIA_EXPRESS_NOTIFY: 43,
                    UNDECRYPTED_MEDIA: 44,
                    UNDECRYPTED_TEXT: 45,
                    UNDECRYPTED_PAY: 46,
                    UNDECRYPTED_POLL: 47,
                    UNDECRYPTED_REACTION: 48,
                    UNDECRYPTED_MEDIA_NOTIFY: 49,
                    VIEW_ONCE_AUDIO: 50,
                    SCHEDULED_CALL_CREATE: 51,
                    SCHEDULED_CALL_CANCEL: 52,
                    PUSH_TO_VIDEO: 53,
                    PIN_IN_CHAT: 54,
                    INTERACTIVE_CAROUSEL: 55,
                    COMMENT: 56,
                    EPHEMERAL_SYNC_RESPONSE: 57,
                    EVENT_CREATE: 58,
                    EVENT_RESPOND: 59,
                    LOTTIE_STICKER: 60,
                    INTERACTIVE_PRODUCT_CAROUSEL: 61,
                    INTERACTIVE_PRODUCT: 62,
                },
            ],
            messageQueueTime: [15, 'timer'],
            messageReceiveT0: [6, 'timer'],
            messageReceiveT1: [7, 'timer'],
            messageType: [
                1,
                {
                    INDIVIDUAL: 1,
                    GROUP: 2,
                    BROADCAST: 3,
                    STATUS: 4,
                    CHANNEL: 5,
                    INTEROP: 6,
                },
            ],
            mutedGroupMessage: [8, 'boolean'],
            numOfWebUrlsInTextMessage: [3, 'integer'],
            offlineCount: [30, 'integer'],
            oppositeVisibleIdentification: [
                40,
                {
                    PHONE_NUMBER: 1,
                    SAVED_CONTACT_NAME: 2,
                    USERNAME: 3,
                    MASKED_PHONE_NUMBER: 4,
                    VERIFIED_BUSINESS_NAME: 5,
                    PLACEHOLDER: 6,
                    PUSHNAME: 7,
                },
            ],
            paddingBytesSize: [22, 'integer'],
            participantCount: [17, 'integer'],
            receiverDefaultDisappearingDuration: [12, 'integer'],
            revokeType: [
                20,
                {
                    SENDER: 0,
                    ADMIN: 1,
                },
            ],
            senderDefaultDisappearingDuration: [11, 'integer'],
            serverAddressingMode: [
                35,
                {
                    PN: 1,
                    LID: 2,
                },
            ],
            stickerIsAi: [29, 'boolean'],
            stickerIsFromStickerMaker: [31, 'boolean'],
            stickerMakerSourceType: [
                32,
                {
                    CUTOUT_IMAGE: 1,
                    WEB_STICKER_MAKER: 2,
                    IOS_STICKER_MAKER: 3,
                    ANDROID_STICKER_MAKER: 4,
                    TRANSPARENT_IMAGE: 5,
                },
            ],
            typeOfGroup: [
                21,
                {
                    GROUP: 1,
                    SUBGROUP: 2,
                    DEFAULT_SUBGROUP: 3,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'PlaceholderActivity',
        id: 1980,
        props: {
            deviceCount: [9, 'integer'],
            deviceSizeBucket: [
                8,
                {
                    LT32: 1,
                    LT64: 2,
                    LT128: 3,
                    LT256: 4,
                    LT512: 5,
                    LT1000: 6,
                    LT1024: 16,
                    LT1500: 7,
                    LT2000: 8,
                    LT2500: 9,
                    LT3000: 10,
                    LT3500: 11,
                    LT4000: 12,
                    LT4500: 13,
                    LT5000: 14,
                    LARGEST_BUCKET: 15,
                },
            ],
            e2eSenderType: [
                16,
                {
                    MY_PRIMARY: 1,
                    OTHER_PRIMARY: 2,
                    MY_COMPANION: 3,
                    OTHER_COMPANION: 4,
                    MY_HOSTED_COMPANION: 5,
                    OTHER_HOSTED_COMPANION: 6,
                },
            ],
            isHostedChat: [19, 'boolean'],
            isLid: [12, 'boolean'],
            messageBeforeReg: [15, 'boolean'],
            messageIsRevoke: [7, 'boolean'],
            messageKeyHash: [14, 'string'],
            messageMediaType: [
                6,
                {
                    NONE: 1,
                    PHOTO: 2,
                    VIDEO: 3,
                    AUDIO: 4,
                    PTT: 5,
                    LOCATION: 6,
                    CONTACT: 7,
                    DOCUMENT: 8,
                    URL: 9,
                    CALL: 10,
                    GIF: 11,
                    FUTURE: 12,
                    CONTACT_ARRAY: 13,
                    LIVE_LOCATION: 14,
                    PROFILE_PIC: 15,
                    STICKER: 16,
                    HSM: 17,
                    PRODUCT_IMAGE: 18,
                    TEMPLATE: 19,
                    MD_APP_STATE: 20,
                    MD_HISTORY_SYNC: 21,
                    CATALOG_LINK: 22,
                    PRODUCT_LINK: 23,
                    ORDER: 24,
                    LIST: 25,
                    LIST_REPLY: 26,
                    BUTTON_MESSAGE: 27,
                    BUTTON_RESPONSE_MESSAGE: 28,
                    PRODUCT_LIST: 29,
                    BUTTON_NFM: 30,
                    VIEW_ONCE_IMAGE: 31,
                    VIEW_ONCE_VIDEO: 32,
                    SHOP_STOREFRONT: 33,
                    REACTION: 34,
                    INTERACTIVE_NFM: 35,
                    INTERACTIVE_RESPONSE_NFM: 36,
                    POLL_CREATE: 37,
                    POLL_VOTE: 38,
                    TEXT: 39,
                    KEEP: 40,
                    UNKEEP: 41,
                    INVISIBLE_HELLO: 42,
                    MEDIA_EXPRESS_NOTIFY: 43,
                    UNDECRYPTED_MEDIA: 44,
                    UNDECRYPTED_TEXT: 45,
                    UNDECRYPTED_PAY: 46,
                    UNDECRYPTED_POLL: 47,
                    UNDECRYPTED_REACTION: 48,
                    UNDECRYPTED_MEDIA_NOTIFY: 49,
                    VIEW_ONCE_AUDIO: 50,
                    SCHEDULED_CALL_CREATE: 51,
                    SCHEDULED_CALL_CANCEL: 52,
                    PUSH_TO_VIDEO: 53,
                    PIN_IN_CHAT: 54,
                    INTERACTIVE_CAROUSEL: 55,
                    COMMENT: 56,
                    EPHEMERAL_SYNC_RESPONSE: 57,
                    EVENT_CREATE: 58,
                    EVENT_RESPOND: 59,
                    LOTTIE_STICKER: 60,
                    INTERACTIVE_PRODUCT_CAROUSEL: 61,
                    INTERACTIVE_PRODUCT: 62,
                },
            ],
            messageType: [
                5,
                {
                    INDIVIDUAL: 1,
                    GROUP: 2,
                    BROADCAST: 3,
                    STATUS: 4,
                    CHANNEL: 5,
                    INTEROP: 6,
                },
            ],
            participantCount: [10, 'integer'],
            placeholderActionInd: [
                2,
                {
                    OTHER: 0,
                    ADD: 1,
                    VIEW: 2,
                    POPULATE: 3,
                },
            ],
            placeholderAddReason: [
                11,
                {
                    OTHER: 0,
                    SIGNAL_NO_SESSION: 1,
                    DEVICE_VERIFICATION_FAILURE: 2,
                    UNKNOWN_SELF_DEVICE: 3,
                    SIGNAL_INVALID_KEY: 4,
                    SIGNAL_INVALID_KEY_ID: 5,
                    SIGNAL_INVALID_MESSAGE: 6,
                    DEVICE_VERIFICATION_FAILURE_SELF_PEER: 7,
                    UNKNOWN_COMPANION_NO_PREKEY: 8,
                    BAD_EPHEMERAL_SETTING: 9,
                    SIGNAL_FUTURE_MESSAGE: 10,
                    SIGNAL_INVALID_SIGNATURE: 11,
                    SIGNAL_BAD_MAC: 12,
                    SIGNAL_INVALID_SESSION: 13,
                },
            ],
            placeholderChatTypeInd: [
                3,
                {
                    OTHER: 0,
                    INDIVIDUAL: 1,
                    GROUP: 2,
                    STATUS: 3,
                    BROADCAST: 4,
                    CHANNEL: 5,
                    INTEROP: 6,
                },
            ],
            placeholderPopulationType: [
                17,
                {
                    OTHER: 0,
                    RETRY: 1,
                    PEER_MESSAGE: 2,
                    RESEND: 3,
                },
            ],
            placeholderTimePeriod: [4, 'integer'],
            placeholderTypeInd: [
                1,
                {
                    OTHER: 0,
                    CIPHERTEXT: 1,
                    FANOUT: 2,
                    DOWNGRADE: 3,
                },
            ],
            typeOfGroup: [
                13,
                {
                    GROUP: 1,
                    SUBGROUP: 2,
                    DEFAULT_SUBGROUP: 3,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'SenderKeyExpired',
        id: 3130,
        props: {
            chatType: [
                1,
                {
                    OTHER: 0,
                    INDIVIDUAL: 1,
                    GROUP: 2,
                    STATUS: 3,
                    BROADCAST: 4,
                    CHANNEL: 5,
                },
            ],
            deviceSizeBucket: [
                2,
                {
                    LT32: 1,
                    LT64: 2,
                    LT128: 3,
                    LT256: 4,
                    LT512: 5,
                    LT1000: 6,
                    LT1024: 16,
                    LT1500: 7,
                    LT2000: 8,
                    LT2500: 9,
                    LT3000: 10,
                    LT3500: 11,
                    LT4000: 12,
                    LT4500: 13,
                    LT5000: 14,
                    LARGEST_BUCKET: 15,
                },
            ],
            expiryReason: [
                3,
                {
                    USER_LEAVE: 1,
                    DEVICE_UNPAIR: 2,
                    IDENTITY_CHANGE: 3,
                    AUDIENCE_CHANGE: 4,
                    PERIODIC_ROTATION: 5,
                    KEY_CORRUPTION: 6,
                    PEER_COMPANION_UNPAIR: 7,
                    OTHER_DEVICE_UNPAIR: 8,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MexEventV2',
        id: 4336,
        props: {
            mexEventV2DurationMs: [1, 'timer'],
            mexEventV2EndTime: [2, 'integer'],
            mexEventV2ErrorCodes: [3, 'string'],
            mexEventV2Errors: [4, 'string'],
            mexEventV2HasData: [5, 'boolean'],
            mexEventV2IsMex: [6, 'boolean'],
            mexEventV2OperationName: [7, 'string'],
            mexEventV2QueryId: [8, 'string'],
            mexEventV2StartTime: [9, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'GraphqlCatalogRequest',
        id: 3206,
        props: {
            businessJid: [4, 'string'],
            businessType: [
                5,
                {
                    SMB: 1,
                    API_DC: 2,
                    API: 3,
                },
            ],
            graphqlCatalogEndpoint: [
                1,
                {
                    GET_CATALOG: 1,
                    GET_PRODUCT: 2,
                    GET_PRODUCT_LIST: 3,
                    GET_COLLECTIONS: 4,
                    GET_SINGLE_COLLECTION: 5,
                    GET_CATEGORIES: 6,
                    GET_VARIANTS: 7,
                    GET_PROMOTIONS: 8,
                },
            ],
            graphqlErrorCode: [3, 'integer'],
            graphqlRequestResult: [
                2,
                {
                    SUCCESS: 1,
                    FAILURE: 2,
                },
            ],
        },
        weight: 1,
        wamChannel: 'private',
        privateStatsIdInt: 0,
    },
    {
        name: 'MexEvent',
        id: 3782,
        props: {
            isMex: [1, 'boolean'],
            mexEventData: [2, 'string'],
            mexEventDurationT: [12, 'timer'],
            mexEventEndTime: [3, 'integer'],
            mexEventEnvelopeResponseStatus: [13, 'integer'],
            mexEventOperation: [7, 'string'],
            mexEventPayloadResponseStatus: [14, 'integer'],
            mexEventRequestSize: [8, 'integer'],
            mexEventResponseSize: [9, 'integer'],
            mexEventRetries: [10, 'integer'],
            mexEventStartTime: [11, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'FmxAction',
        id: 4444,
        props: {
            commonGroupNum: [1, 'integer'],
            countryShown: [2, 'boolean'],
            fmxEntryPoint: [
                3,
                {
                    FMX_CARD: 0,
                    SAFETY_TOOLS: 1,
                },
            ],
            fmxEvent: [
                4,
                {
                    BLOCK: 0,
                    ADD_CONTACT: 1,
                    REPORT: 2,
                    SAFETY_TOOLS: 3,
                    COMMON_GROUPS: 4,
                    CONTACT_INFO: 5,
                    FMX_CARD_INSERTED: 6,
                    FMX_CARD_VIEWED: 7,
                    LEARN_MORE: 8,
                    HIGHLIGHT_GROUP_NAME: 9,
                },
            ],
            highlightGroupType: [
                6,
                {
                    UNKNOWN: 0,
                    SINGLE: 1,
                    CREATOR: 2,
                    ADMIN: 3,
                    SAVED_CONTACTS: 4,
                    PARTICIPANTS: 5,
                    MORE: 6,
                },
            ],
            isSenderSmb: [7, 'boolean'],
            notAContactShown: [5, 'boolean'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'UnknownStanza',
        id: 3448,
        props: {
            unknownStanzaTag: [1, 'string'],
            unknownStanzaType: [2, 'string'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'IncomingMessageDrop',
        id: 3724,
        props: {
            agentEngagementType: [
                12,
                {
                    DIRECT_CHAT: 0,
                    INVOKED: 1,
                    MEMBER: 2,
                },
            ],
            botType: [
                14,
                {
                    UNKNOWN: 0,
                    METABOT: 1,
                    BOT_1P_BIZ: 2,
                    BOT_3P_BIZ: 3,
                },
            ],
            e2eCiphertextType: [
                1,
                {
                    MESSAGE: 0,
                    PREKEY_MESSAGE: 1,
                    SENDER_KEY_MESSAGE: 2,
                    MESSAGE_SECRET_MESSAGE: 3,
                },
            ],
            e2eDestination: [
                2,
                {
                    INDIVIDUAL: 0,
                    GROUP: 1,
                    LIST: 2,
                    STATUS: 3,
                    CHANNEL: 4,
                    INTEROP: 5,
                },
            ],
            e2eFailureReason: [
                9,
                {
                    INVALID_MESSAGE: 0,
                    LEGACY_MESSAGE: 1,
                    INVALID_VERSION: 2,
                    PRE_KEY_MESSAGE_MISSING_PRE_KEY: 3,
                    PRE_KEY_MESSAGE_INVALID_KEY: 4,
                    UNTRUSTED_IDENTITY: 5,
                    NO_SESSION_AVAILABLE: 6,
                    PLAINTEXT_BYTE_MISMATCH: 7,
                    UNKNOWN_CIPHERTEXT_TYPE: 8,
                    UNKNOWN_CIPHERTEXT_VERSION: 9,
                    INVALID_PADDING: 10,
                    INVALID_PROTOCOL_BUFFER: 11,
                    INVALID_NUMBER_OF_MESSAGE_TYPES: 12,
                    INVALID_IMAGE_FILE_LENGTH: 13,
                    INVALID_IMAGE_FILE_SHA256: 14,
                    INVALID_IMAGE_MEDIA_URL: 15,
                    INVALID_IMAGE_MEDIA_KEY: 16,
                    INVALID_IMAGE_MIME_TYPE: 17,
                    UNKNOWN_V2_TAGS: 18,
                    INVALID_EDIT_VERSION: 19,
                    INVALID_HSM_NAMESPACE: 20,
                    INVALID_HSM_ELEMENT: 21,
                    UNKNOWN_PROTOCOL_MESSAGE_TYPE: 22,
                    DUPLICATE_MESSAGE: 23,
                    FINGERPRINT_IDENTITY_MISMATCH: 24,
                    FINGERPRINT_VERSION_MISMATCH: 25,
                    INVALID_ARGUMENT: 26,
                    INVALID_KEY: 27,
                    INVALID_KEY_ID: 28,
                    INVALID_MAC: 29,
                    NO_MEMORY: 30,
                    STALE_KEY_EXCHANGE: 31,
                    VRF_SIGNATURE_VERIFICATION_FAILED: 32,
                    ERROR_MINIMUM_VALUE: 33,
                    ERROR_UNKNOWN: 34,
                    INVALID_BROADCAST_STANZA_ATTRIBUTE: 35,
                    INVALID_LIST_REPLY_MESSAGE: 36,
                    INVALID_LIST_REPLY_ROW_IDENTIFIER: 37,
                    INVALID_BUTTONS_RESPONSE_MESSAGE: 38,
                    INVALID_BUTTONS_RESPONSE_BUTTON_IDENTIFIER: 39,
                    INVALID_ORDER_MESSAGE: 40,
                    ERROR_CANCELLED: 41,
                    FUTURE_MESSAGE: 42,
                    INVALID_EMPTY_PROTOBUF_NOT_ALLOWED: 43,
                    SELF_SEND: 44,
                    INVALID_MEDIA_MIME_TYPE_TEMPLATE_MESSAGE: 45,
                    INVALID_MEDIA_ENC_HASH: 46,
                    INVALID_IDENTITY: 47,
                    PASSTHROUGH: 48,
                    NOT_IN_PENDING_DEVICES: 49,
                    EDIT_TYPE_AND_PROTOCOL_MISMATCH: 50,
                    DECRYPTION_FAILED: 51,
                    DESERIALIZE_INVALID_PROTO_FORMAT: 52,
                    DESERIALIZE_RATCHET_KEY_BAD_FORMAT: 53,
                    DESERIALIZE_PK_INVALID_PROTO_FORMAT: 54,
                    DESERIALZE_PK_KEY_BAD_FORMAT: 55,
                    TOO_MANY_MESSAGES_IN_FUTURE: 56,
                    EMPTY_VERSION_CONTENT_SUFFIX: 57,
                    CONTENT_EXCEEDED_EXPECTED_LENGTH: 58,
                    GRP_INVALID_PROTO_FORMAT: 59,
                    GRP_INVALID_KEY_FORMAT: 60,
                    GRP_VERSION_CONTENT_EMPTY: 61,
                    GRP_INVALID_VERSION_CONTENT_LENGTH: 62,
                    GRP_SENDER_KEY_INVALID_PROTO_FORMAT: 63,
                    GRP_SENDER_KEY_PROTO_ERROR: 64,
                    GRP_TOO_MANY_MESSAGES_IN_FUTURE: 65,
                    MISSING_MESSAGE_SECRET: 66,
                    MESSAGE_SECRET_INVALID: 67,
                    INVALID_REACTION_TEXT: 68,
                    UNSUPPORTED_MESSAGE_TYPE: 69,
                    UNSUPPORTED_IN_INTEROP: 70,
                    TARGET_MESSAGE_NOT_FOUND: 71,
                    STATUS_FROM_HOSTED_DEVICE_UNSUPPORTED: 72,
                    GROUP_MESSAGE_FROM_HOSTED_DEVICE_UNSUPPORTED: 73,
                    BROADCAST_FROM_HOSTED_DEVICE_UNSUPPORTED: 74,
                    LIVE_LOCATION_FROM_HOSTED_DEVICE_UNSUPPORTED: 75,
                    MESSAGE_TYPE_MISMATCH: 76,
                    ERROR_SESSION_STATE_UNACKED_PKMSG_GET_BASE_KEY: 77,
                    ERROR_SIGNAL_MESSAGE_GET_SENDER_RATCHET_KEY: 78,
                    ERROR_SESSION_STATE_GET_ROOT_KEY: 79,
                    ERROR_SESSION_STATE_GET_SENDER_RATCHET_KEY_PAIR: 80,
                    ERROR_SESSION_STATE_SET_SENDER_CHAIN_KEY: 81,
                    ERROR_SESSION_STATE_FIND_RECEIVER_CHAIN: 82,
                    ERROR_SESSION_RECORD_DESERIALIZE: 83,
                    ERROR_SENDER_KEY_STATE_REMOVE_SENDER_MESSAGE_KEY: 84,
                    ERROR_EMPTY_PLAINTEXT_DATA: 85,
                    ERROR_MAC_LEN_MISMATCH: 86,
                    ERROR_INVALID_MAC_LEN: 87,
                    ERROR_INVALID_SIGNATURE_LEN: 88,
                    ERROR_SENDER_KEY_LOAD_KEY: 89,
                    ERROR_FAST_RATCHET_SENDER_KEY_LOAD_KEY: 90,
                    ERROR_UNKNOWN_LOAD_KEY: 91,
                    ERROR_UNKNOWN_LOAD_SESSION: 92,
                    ERROR_SESSION_RECORD_GET_STATE: 93,
                    ERROR_SESSION_STATE_GET_SENDER_CHAIN_KEY: 94,
                    ERROR_INVALID_KEY_MATEIRAL_DATA_LEN: 95,
                    ERROR_SESSION_STATE_GET_SENDER_RATCHET_KEY: 96,
                    ERROR_SESSION_STATE_GET_LOCAL_IDENTITY_KEY: 97,
                    ERROR_SESSION_STATE_GET_REMOTE_IDENTITY_KEY: 98,
                },
            ],
            e2eSenderType: [
                3,
                {
                    MY_PRIMARY: 1,
                    OTHER_PRIMARY: 2,
                    MY_COMPANION: 3,
                    OTHER_COMPANION: 4,
                    MY_HOSTED_COMPANION: 5,
                    OTHER_HOSTED_COMPANION: 6,
                },
            ],
            invisibleMessageCategory: [
                13,
                {
                    PEER: 1,
                    INVISIBLE_KEY_DISTRIBUTION: 2,
                    OTHER: 3,
                },
            ],
            messageDropReason: [
                4,
                {
                    SYNCD_DELETION: 1,
                    ADMIN_REVOKE_NOT_ENABLED: 2,
                    RECEIVED_WITH_OLD_COUNTER: 3,
                    INVALID_STANZA: 4,
                    INVALID_PROTOBUF: 5,
                    MESSAGE_SECRET_ERROR: 6,
                    INVALID_LID_ADDRESSED_MESSAGE: 7,
                    UNKNOWN_MESSAGE_TYPE: 8,
                    DB_OPERATION_FAILED: 9,
                    INTERNAL_ERROR: 10,
                    EXPIRED: 11,
                    INVALID_HOSTED_COMPANION_STANZA: 12,
                    MESSAGE_REVOKED: 13,
                    PAYMENT_MESSAGE_REVOKED: 14,
                    DUPLICATE_MESSAGE: 15,
                    DUPLICATE_DELIVERY: 16,
                },
            ],
            messageMediaType: [
                5,
                {
                    NONE: 1,
                    PHOTO: 2,
                    VIDEO: 3,
                    AUDIO: 4,
                    PTT: 5,
                    LOCATION: 6,
                    CONTACT: 7,
                    DOCUMENT: 8,
                    URL: 9,
                    CALL: 10,
                    GIF: 11,
                    FUTURE: 12,
                    CONTACT_ARRAY: 13,
                    LIVE_LOCATION: 14,
                    PROFILE_PIC: 15,
                    STICKER: 16,
                    HSM: 17,
                    PRODUCT_IMAGE: 18,
                    TEMPLATE: 19,
                    MD_APP_STATE: 20,
                    MD_HISTORY_SYNC: 21,
                    CATALOG_LINK: 22,
                    PRODUCT_LINK: 23,
                    ORDER: 24,
                    LIST: 25,
                    LIST_REPLY: 26,
                    BUTTON_MESSAGE: 27,
                    BUTTON_RESPONSE_MESSAGE: 28,
                    PRODUCT_LIST: 29,
                    BUTTON_NFM: 30,
                    VIEW_ONCE_IMAGE: 31,
                    VIEW_ONCE_VIDEO: 32,
                    SHOP_STOREFRONT: 33,
                    REACTION: 34,
                    INTERACTIVE_NFM: 35,
                    INTERACTIVE_RESPONSE_NFM: 36,
                    POLL_CREATE: 37,
                    POLL_VOTE: 38,
                    TEXT: 39,
                    KEEP: 40,
                    UNKEEP: 41,
                    INVISIBLE_HELLO: 42,
                    MEDIA_EXPRESS_NOTIFY: 43,
                    UNDECRYPTED_MEDIA: 44,
                    UNDECRYPTED_TEXT: 45,
                    UNDECRYPTED_PAY: 46,
                    UNDECRYPTED_POLL: 47,
                    UNDECRYPTED_REACTION: 48,
                    UNDECRYPTED_MEDIA_NOTIFY: 49,
                    VIEW_ONCE_AUDIO: 50,
                    SCHEDULED_CALL_CREATE: 51,
                    SCHEDULED_CALL_CANCEL: 52,
                    PUSH_TO_VIDEO: 53,
                    PIN_IN_CHAT: 54,
                    INTERACTIVE_CAROUSEL: 55,
                    COMMENT: 56,
                    EPHEMERAL_SYNC_RESPONSE: 57,
                    EVENT_CREATE: 58,
                    EVENT_RESPOND: 59,
                    LOTTIE_STICKER: 60,
                    INTERACTIVE_PRODUCT_CAROUSEL: 61,
                    INTERACTIVE_PRODUCT: 62,
                },
            ],
            offline: [6, 'boolean'],
            offlineCount: [11, 'integer'],
            retryCount: [7, 'integer'],
            revokeType: [
                8,
                {
                    SENDER: 0,
                    ADMIN: 1,
                },
            ],
            typeOfGroup: [
                10,
                {
                    GROUP: 1,
                    SUBGROUP: 2,
                    DEFAULT_SUBGROUP: 3,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebcOfflineNotificationProcess',
        id: 4222,
        props: {
            currentOfflineProcessStage: [
                4,
                {
                    PAGE_LOAD: 1,
                    SOCKET_CONNECT: 2,
                    OFFLINE_PREVIEW: 3,
                    PROCESSING: 4,
                    PROCESS_COMPLETE: 5,
                    PROCESS_INTERRUPTED: 6,
                },
            ],
            offlineProcessDecryptErrorCount: [5, 'integer'],
            offlineProcessMailboxAge: [6, 'integer'],
            offlineProcessMessageCount: [7, 'integer'],
            offlineProcessNotificationCount: [8, 'integer'],
            offlineProcessSessionId: [9, 'string'],
            offlineProcessStageTimestampMs: [10, 'integer'],
            runReason: [
                12,
                {
                    PUSH_NOTIFICATION: 1,
                    PERIODIC_BACKGROUND_SYNC: 2,
                },
            ],
            swVersion: [11, 'string'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MdCriticalEvent',
        id: 2746,
        props: {
            collection: [
                2,
                {
                    REGULAR: 1,
                    REGULAR_LOW: 2,
                    REGULAR_HIGH: 3,
                    CRITICAL_BLOCK: 4,
                    CRITICAL_UNBLOCK_LOW: 5,
                },
            ],
            mdCriticalEventCode: [
                1,
                {
                    MESSAGE_RANGE_UNSET: 1,
                    MESSAGE_RANGE_LAST_SYSTEM_MESSAGE_TIMESTAMP_SET: 2,
                    MESSAGE_RANGE_MESSAGES_UNSET: 3,
                    MESSAGE_RANGE_MESSAGES_EMPTY: 4,
                    MESSAGE_RANGE_MESSAGES_CROSS_LIMIT: 5,
                    MESSAGE_RANGE_MESSAGE_KEY_UNSET: 6,
                    MESSAGE_RANGE_MESSAGE_KEY_REMOTE_JID_UNSET: 7,
                    MESSAGE_RANGE_MESSAGE_KEY_FROM_ME_UNSET: 8,
                    MESSAGE_RANGE_MESSAGE_KEY_STANZA_ID_UNSET: 9,
                    MESSAGE_RANGE_MESSAGE_KEY_REMOTE_JID_INVALID: 10,
                    MESSAGE_RANGE_MESSAGE_KEY_PARTICIPANT_UNSET: 11,
                    MALFORMED_PENDING_MUTATION: 12,
                    ACTION_INVALID_INDEX_DATA: 13,
                    MISSING_MUTATION_TO_REMOVE: 14,
                    LTHASH_INCONSISTENCY_ON_DAILY_CHECK: 15,
                    LTHASH_INCONSISTENCY_ON_SNAPSHOT_MAC_MISMATCH: 16,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MdLinkDeviceCompanion',
        id: 2576,
        props: {
            mdDurationS: [8, 'integer'],
            mdLinkDeviceCompanionErrorCode: [6, 'integer'],
            mdLinkDeviceCompanionStage: [
                4,
                {
                    PAIR_SUCCESS_RECEIVED: 1,
                    PAIR_DEVICE_SIGN_SENT: 2,
                    FIRST_CONNECT: 3,
                    UPLOAD_PREKEYS: 4,
                    COMPLETE: 5,
                    GENERATE_PREKEYS: 6,
                    SENT_PREKEYS: 7,
                },
            ],
            mdRegAttemptId: [9, 'string'],
            mdSessionId: [1, 'string'],
            mdTimestampS: [7, 'integer'],
            mdWasUpgraded: [5, 'boolean'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'AdvStoredTimestampExpired',
        id: 3036,
        props: {
            advExpireTimeInHours: [1, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebcQplHealth',
        id: 3134,
        props: {
            webcQplHealthEventData: [1, 'string'],
            webcQplHealthEventType: [
                2,
                {
                    ANNOTATION_SIZE_LIMIT_EXCEEDED: 1,
                    MAX_POINT_COUNT_EXCEEDED: 2,
                    MAX_MARKER_COUNT_EXCEEDED: 3,
                    TOO_MANY_OPEN_MARKERS_TO_WRITE: 4,
                    POINT_TO_END_AT_NOT_FOUND: 5,
                    JSON_FORMAT_ERROR: 6,
                    MAX_STORAGE_EVENT_COUNT_REACHED: 7,
                    ERROR_UPLOADING_CHUNK: 8,
                    POINT_NAME_TOO_LONG: 9,
                    ANNOTATION_KEY_TOO_LONG: 10,
                    POINT_DATA_TOO_LONG: 11,
                    ERROR_PARSING_CONFIG: 12,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'CommunityHomeAction',
        id: 3494,
        props: {
            communityHomeGroupDiscoveries: [1, 'integer'],
            communityHomeGroupJoins: [2, 'integer'],
            communityHomeGroupNavigations: [3, 'integer'],
            communityHomeId: [4, 'string'],
            communityHomeViews: [5, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'CommunityTabAction',
        id: 3496,
        props: {
            communityNoActionTabViews: [4, 'integer'],
            communityTabGroupNavigations: [1, 'integer'],
            communityTabToHomeViews: [2, 'integer'],
            communityTabViews: [3, 'integer'],
            communityTabViewsViaContextMenu: [5, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'Daily',
        id: 1158,
        props: {
            accessibilityVoiceover: [108, 'boolean'],
            addressbookSize: [11, 'integer'],
            addressbookWhatsappSize: [12, 'integer'],
            androidAdvertisingId: [135, 'string'],
            androidKeystoreState: [
                139,
                {
                    NOT_AVAILABLE: 1,
                    SELF_TEST_FAILURE: 2,
                    ENC_KEY_READ_FAILURE: 3,
                    ENC_KEY_STORED_USED: 4,
                    ENC_KEY_PLAIN_DELETED: 5,
                    ENC_KEY_PLAIN_RECOVERED: 6,
                },
            ],
            androidXmppWorkersRuntime: [167, 'integer'],
            appCodeHash: [103, 'string'],
            appStandbyBucket: [121, 'integer'],
            autoDlAudioCellular: [90, 'boolean'],
            autoDlAudioRoaming: [91, 'boolean'],
            autoDlAudioWifi: [89, 'boolean'],
            autoDlDocCellular: [96, 'boolean'],
            autoDlDocRoaming: [97, 'boolean'],
            autoDlDocWifi: [95, 'boolean'],
            autoDlImageCellular: [87, 'boolean'],
            autoDlImageRoaming: [88, 'boolean'],
            autoDlImageWifi: [86, 'boolean'],
            autoDlVideoCellular: [93, 'boolean'],
            autoDlVideoRoaming: [94, 'boolean'],
            autoDlVideoWifi: [92, 'boolean'],
            backupNetworkSetting: [
                10,
                {
                    WIFI_ONLY: 0,
                    WIFI_OR_CELLULAR: 1,
                },
            ],
            backupRestoreEncryptionVersion: [138, 'integer'],
            backupSchedule: [
                9,
                {
                    OFF: 0,
                    DAILY: 1,
                    WEEKLY: 2,
                    MONTHLY: 3,
                    MANUAL: 4,
                },
            ],
            channelsMediaFolderSize: [186, 'integer'],
            chatDatabaseSize: [19, 'integer'],
            chatLockFolderCount: [168, 'integer'],
            countEphemeralThreads: [181, 'integer'],
            countEphemeralThreadsEnabledByMe: [182, 'integer'],
            defaultDisappearingDuration: [140, 'integer'],
            deviceLanguage: [153, 'string'],
            emailState: [
                174,
                {
                    NOT_ADDED: 1,
                    UNVERIFIED: 2,
                    VERIFIED: 3,
                },
            ],
            entSecurityNotificationsEnabled: [134, 'boolean'],
            experimentTmoPreloadGroupDaily: [166, 'integer'],
            favoritedAnimatedStickerCount: [113, 'integer'],
            favoritedFirstPartyStickerCount: [112, 'integer'],
            favoritedTotalStickerCount: [111, 'integer'],
            hasTextstatusEmojiModified24h: [175, 'boolean'],
            hasTextstatusModified24h: [176, 'boolean'],
            hasTextstatusTextModified24h: [177, 'boolean'],
            hasUsername: [187, 'boolean'],
            installedAnimatedStickerPackCount: [116, 'integer'],
            installedAnimatedThirdPartyStickerPackCount: [137, 'integer'],
            installedFirstPartyStickerPackCount: [115, 'integer'],
            installedTotalStickerPackCount: [114, 'integer'],
            ipStr: [122, 'string'],
            isDefaultDisappearingMessagingUser: [183, 'boolean'],
            isEphemeralMessagingUser: [184, 'boolean'],
            keyboardLanguage: [154, 'string'],
            languageCode: [5, 'string'],
            lastCloudBackupSize: [185, 'integer'],
            locationCode: [6, 'string'],
            lockFolderHidden: [171, 'boolean'],
            lowestAppStandbyBucket: [160, 'integer'],
            mdPairTime: [124, 'integer'],
            mediaFolderFileCount: [21, 'integer'],
            mediaFolderSize: [20, 'integer'],
            modifiedInternalProps: [155, 'boolean'],
            networkIsRoaming: [7, 'boolean'],
            networkOperatorName: [159, 'string'],
            numAccounts: [163, 'integer'],
            numContactsWithTextstatus24h: [178, 'integer'],
            numContactsWithTextstatusEmoji24h: [179, 'integer'],
            numContactsWithTextstatusText24h: [180, 'integer'],
            osBuildNumber: [4, 'string'],
            osNotificationSetting: [
                118,
                {
                    ALLOWED: 1,
                    BLOCKED: 2,
                    UNKNOWN: 3,
                },
            ],
            packageName: [102, 'string'],
            passkeyExists: [165, 'boolean'],
            paymentsIsEnabled: [100, 'boolean'],
            permissionAccessCoarseLocation: [57, 'integer'],
            permissionAccessFineLocation: [58, 'integer'],
            permissionCamera: [56, 'integer'],
            permissionContacts: [104, 'boolean'],
            permissionReadExternalStorage: [53, 'integer'],
            permissionRecordAudio: [55, 'integer'],
            phoneCores: [156, 'integer'],
            phoneyid: [162, 'string'],
            privacySettingsAbout: [
                141,
                {
                    NOBODY: 1,
                    ONLY_SHARE_WITH: 2,
                    MY_CONTACTS: 3,
                    MY_CONTACTS_EXCEPT: 4,
                    EVERYONE: 5,
                },
            ],
            privacySettingsAboutExceptNum: [
                142,
                {
                    B0: 1,
                    B1: 2,
                    B5: 3,
                    B10: 4,
                    B15: 5,
                    B20: 6,
                    B30: 7,
                    B40: 8,
                    B50: 9,
                    B60: 10,
                    B70: 11,
                    B80: 12,
                    B90: 13,
                    B100: 14,
                },
            ],
            privacySettingsGroups: [
                143,
                {
                    NOBODY: 1,
                    ONLY_SHARE_WITH: 2,
                    MY_CONTACTS: 3,
                    MY_CONTACTS_EXCEPT: 4,
                    EVERYONE: 5,
                },
            ],
            privacySettingsGroupsExceptNum: [
                144,
                {
                    B0: 1,
                    B1: 2,
                    B5: 3,
                    B10: 4,
                    B15: 5,
                    B20: 6,
                    B30: 7,
                    B40: 8,
                    B50: 9,
                    B60: 10,
                    B70: 11,
                    B80: 12,
                    B90: 13,
                    B100: 14,
                },
            ],
            privacySettingsLastSeen: [
                145,
                {
                    NOBODY: 1,
                    ONLY_SHARE_WITH: 2,
                    MY_CONTACTS: 3,
                    MY_CONTACTS_EXCEPT: 4,
                    EVERYONE: 5,
                },
            ],
            privacySettingsLastSeenExceptNum: [
                146,
                {
                    B0: 1,
                    B1: 2,
                    B5: 3,
                    B10: 4,
                    B15: 5,
                    B20: 6,
                    B30: 7,
                    B40: 8,
                    B50: 9,
                    B60: 10,
                    B70: 11,
                    B80: 12,
                    B90: 13,
                    B100: 14,
                },
            ],
            privacySettingsProfilePhoto: [
                147,
                {
                    NOBODY: 1,
                    ONLY_SHARE_WITH: 2,
                    MY_CONTACTS: 3,
                    MY_CONTACTS_EXCEPT: 4,
                    EVERYONE: 5,
                },
            ],
            privacySettingsProfilePhotoExceptNum: [
                148,
                {
                    B0: 1,
                    B1: 2,
                    B5: 3,
                    B10: 4,
                    B15: 5,
                    B20: 6,
                    B30: 7,
                    B40: 8,
                    B50: 9,
                    B60: 10,
                    B70: 11,
                    B80: 12,
                    B90: 13,
                    B100: 14,
                },
            ],
            privacySettingsStatus: [
                150,
                {
                    NOBODY: 1,
                    ONLY_SHARE_WITH: 2,
                    MY_CONTACTS: 3,
                    MY_CONTACTS_EXCEPT: 4,
                    EVERYONE: 5,
                },
            ],
            privacySettingsStatusExceptNum: [
                151,
                {
                    B0: 1,
                    B1: 2,
                    B5: 3,
                    B10: 4,
                    B15: 5,
                    B20: 6,
                    B30: 7,
                    B40: 8,
                    B50: 9,
                    B60: 10,
                    B70: 11,
                    B80: 12,
                    B90: 13,
                    B100: 14,
                },
            ],
            privacySettingsStatusShareNum: [
                152,
                {
                    B0: 1,
                    B1: 2,
                    B5: 3,
                    B10: 4,
                    B15: 5,
                    B20: 6,
                    B30: 7,
                    B40: 8,
                    B50: 9,
                    B60: 10,
                    B70: 11,
                    B80: 12,
                    B90: 13,
                    B100: 14,
                },
            ],
            receiptsEnabled: [8, 'boolean'],
            secretCodeActive: [172, 'boolean'],
            showMetaAiButtonSetting: [173, 'boolean'],
            simMcc: [2, 'integer'],
            simMnc: [3, 'integer'],
            storageAvailSize: [31, 'integer'],
            storageTotalSize: [32, 'integer'],
            supportedDecoders: [169, 'string'],
            supportedEncoders: [170, 'string'],
            videoFolderFileCount: [23, 'integer'],
            videoFolderSize: [22, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'KeepInChatNotif',
        id: 3484,
        props: {
            kicGroupNotificationTaps: [3, 'integer'],
            kicGroupNotifications: [4, 'integer'],
            kicNotificationTaps: [5, 'integer'],
            kicNotifications: [6, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'NotificationSetting',
        id: 3684,
        props: {
            groupReactionNotification: [1, 'boolean'],
            groupShowNotification: [2, 'boolean'],
            groupSoundTone: [
                3,
                {
                    DEFAULT: 1,
                    CUSTOM: 2,
                },
            ],
            inAppNotificationAlertStyle: [
                4,
                {
                    NONE: 1,
                    BANNERS: 2,
                    ALERTS: 3,
                },
            ],
            inAppNotificationSound: [5, 'boolean'],
            inAppNotificationVibrate: [6, 'boolean'],
            messageReactionNotification: [7, 'boolean'],
            messageShowNotification: [8, 'boolean'],
            messageSoundTone: [
                9,
                {
                    DEFAULT: 1,
                    CUSTOM: 2,
                },
            ],
            offlineNotification: [11, 'boolean'],
            showPreview: [10, 'boolean'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'PnhDailyCtwa',
        id: 4280,
        props: {
            matMessagesReceived: [1, 'integer'],
            threadDs: [3, 'string'],
            threadId: [2, 'string'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'PnhDaily',
        id: 3806,
        props: {
            communityId: [1, 'string'],
            mappingMissing: [7, 'integer'],
            pnhIndicatorClicksChat: [2, 'integer'],
            pnhIndicatorClicksInfoScreen: [3, 'integer'],
            reactionDeleteCount: [4, 'integer'],
            reactionOpenTrayCount: [5, 'integer'],
            totalContacts: [8, 'integer'],
            typeOfGroup: [
                6,
                {
                    GROUP: 1,
                    SUBGROUP: 2,
                    DEFAULT_SUBGROUP: 3,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'PrivacyHighlightDaily',
        id: 3522,
        props: {
            dialogAppearCount: [1, 'integer'],
            dialogSelectCount: [2, 'integer'],
            narrativeAppearCount: [3, 'integer'],
            privacyHighlightCategory: [
                4,
                {
                    E2EE: 0,
                },
            ],
            privacyHighlightSurface: [
                5,
                {
                    GOLDEN_BOX_CONTACT: 0,
                    GOLDEN_BOX_GROUP: 1,
                    GOLDEN_BOX_BROADCAST: 2,
                    INFO_SCREEN_CONTACT: 3,
                    INFO_SCREEN_GROUP: 4,
                    INFO_SCREEN_BROADCAST: 5,
                    CALLS_LIST: 6,
                    CHATS_LIST: 7,
                    STATUS_LIST: 8,
                    LINKED_DEVICES_SCREEN: 9,
                    CALLING_SCREEN_AUDIO: 10,
                    CALLING_SCREEN_VIDEO: 11,
                    SPLIT_VIEW_HOME_PLACEHOLDER: 12,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'PttDaily',
        id: 2938,
        props: {
            pttCancelBroadcast: [9, 'integer'],
            pttCancelGroup: [8, 'integer'],
            pttCancelIndividual: [7, 'integer'],
            pttCancelInterop: [42, 'integer'],
            pttCancelNewsletter: [32, 'integer'],
            pttDraftReviewBroadcast: [15, 'integer'],
            pttDraftReviewGroup: [14, 'integer'],
            pttDraftReviewIndividual: [13, 'integer'],
            pttDraftReviewInterop: [43, 'integer'],
            pttDraftReviewNewsletter: [33, 'integer'],
            pttFastplaybackBroadcast: [21, 'integer'],
            pttFastplaybackGroup: [20, 'integer'],
            pttFastplaybackIndividual: [19, 'integer'],
            pttFastplaybackInterop: [44, 'integer'],
            pttFastplaybackNewsletter: [34, 'integer'],
            pttLockBroadcast: [12, 'integer'],
            pttLockGroup: [11, 'integer'],
            pttLockIndividual: [10, 'integer'],
            pttLockInterop: [45, 'integer'],
            pttLockNewsletter: [35, 'integer'],
            pttOutOfChatBroadcast: [29, 'integer'],
            pttOutOfChatGroup: [30, 'integer'],
            pttOutOfChatIndividual: [31, 'integer'],
            pttOutOfChatInterop: [46, 'integer'],
            pttOutOfChatNewsletter: [36, 'integer'],
            pttPausedRecordBroadcast: [22, 'integer'],
            pttPausedRecordGroup: [23, 'integer'],
            pttPausedRecordIndividual: [24, 'integer'],
            pttPausedRecordInterop: [47, 'integer'],
            pttPausedRecordNewsletter: [37, 'integer'],
            pttPlaybackBroadcast: [18, 'integer'],
            pttPlaybackGroup: [17, 'integer'],
            pttPlaybackIndividual: [16, 'integer'],
            pttPlaybackInterop: [48, 'integer'],
            pttPlaybackNewsletter: [38, 'integer'],
            pttRecordBroadcast: [3, 'integer'],
            pttRecordGroup: [2, 'integer'],
            pttRecordIndividual: [1, 'integer'],
            pttRecordInterop: [49, 'integer'],
            pttRecordNewsletter: [39, 'integer'],
            pttSendBroadcast: [6, 'integer'],
            pttSendGroup: [5, 'integer'],
            pttSendIndividual: [4, 'integer'],
            pttSendInterop: [50, 'integer'],
            pttSendNewsletter: [40, 'integer'],
            pttStopTapBroadcast: [25, 'integer'],
            pttStopTapGroup: [26, 'integer'],
            pttStopTapIndividual: [27, 'integer'],
            pttStopTapInterop: [51, 'integer'],
            pttStopTapNewsletter: [41, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'ScreenLockSettingsData',
        id: 4802,
        props: {
            screenAutoLockDuration: [1, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'ScreenLockSettings',
        id: 3872,
        props: {
            screenLockDuration: [1, 'integer'],
        },
        weight: 1,
        wamChannel: 'private',
        privateStatsIdInt: 113760892,
    },
    {
        name: 'WebcFtsStorage',
        id: 3642,
        props: {
            ftsTotalSize: [1, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'NotificationEngagement',
        id: 3796,
        props: {
            avgNotifEngagementT: [1, 'timer'],
            groupSizeBucket: [
                2,
                {
                    SMALL: 1,
                    MEDIUM: 2,
                    LARGE: 3,
                    EXTRA_LARGE: 4,
                    XX_LARGE: 5,
                    XXX_LARGE: 6,
                    LT1024: 16,
                    LT1500: 7,
                    LT2000: 8,
                    LT2500: 9,
                    LT3000: 10,
                    LT3500: 11,
                    LT4000: 12,
                    LT4500: 13,
                    LT5000: 14,
                    LARGEST_BUCKET: 15,
                },
            ],
            groupTypeClient: [
                3,
                {
                    REGULAR_GROUP: 1,
                    SUB_GROUP: 2,
                    DEFAULT_SUB_GROUP: 3,
                    PARENT_GROUP: 4,
                },
            ],
            isAGroup: [4, 'boolean'],
            isWebBackgroundSyncNotif: [18, 'boolean'],
            threadId: [5, 'string'],
            totalMessageReminderNotifShown: [16, 'integer'],
            totalMessageReminderNotifTapToOpen: [17, 'integer'],
            totalNotifMarkAsRead: [6, 'integer'],
            totalNotifMissedCallVoipCallback: [7, 'integer'],
            totalNotifMissedCallVoipMessage: [8, 'integer'],
            totalNotifOthers: [9, 'integer'],
            totalNotifReply: [10, 'integer'],
            totalNotifRtcVoipAccept: [11, 'integer'],
            totalNotifRtcVoipDecline: [12, 'integer'],
            totalNotifShowPreview: [13, 'integer'],
            totalNotifShown: [14, 'integer'],
            totalNotifTapToOpen: [15, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MdBootstrapHistorySyncStatusAfterPairing',
        id: 4652,
        props: {
            activeTimeAfterPairing: [
                1,
                {
                    MINS_10: 1,
                    MINS_20: 2,
                    MINS_40: 3,
                    MINS_60: 4,
                    MINS_5: 5,
                },
            ],
            isLoopRunning: [12, 'boolean'],
            lastProcessedNotificationChunkOrder: [2, 'integer'],
            lastProcessedNotificationChunkProgress: [3, 'integer'],
            mdBootstrapHistoryPayloadType: [
                4,
                {
                    INITIAL: 1,
                    RECENT_HISTORY: 2,
                    FULL_HISTORY: 3,
                    PUSHNAME: 4,
                    STATUS_V3: 5,
                    NON_BLOCKING_DATA: 6,
                    ON_DEMAND: 7,
                },
            ],
            mdHistorySyncStatusResult: [
                5,
                {
                    SUCCESS: 1,
                    UNFINISHED: 2,
                    FAIL_TO_DOWNLOAD: 3,
                    MISSING_CHUNK: 4,
                    PROTOBUF_ERROR: 5,
                    FAIL_TO_STORE: 6,
                    OTHER_ERROR: 7,
                    IN_PROGRESS: 8,
                    FAIL_TO_RECEIVE: 9,
                    FAIL_TO_STORE_CHUNK: 10,
                    FAIL_TO_FETCH: 11,
                    FAIL_TO_PREPROCESS: 12,
                    FAIL_TO_ENCRYPT: 13,
                },
            ],
            mdSessionId: [6, 'string'],
            mdTimestamp: [7, 'integer'],
            missingNotificationCount: [8, 'integer'],
            nextNotificationChunkOrder: [9, 'integer'],
            totalProcessedMessageCount: [10, 'integer'],
            unprocessedNotificationCount: [11, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebDbVersionNonAnonymous',
        id: 4816,
        props: {
            webDbName: [
                1,
                {
                    MODEL_STORAGE: 1,
                    FTS_STORAGE: 2,
                    JOBS_STORAGE: 3,
                    LOADGEN_STORAGE: 4,
                    LRU_MEDIA_STORAGE_IDB: 5,
                    OFFD_STORAGE: 6,
                    QPL_STORAGE: 7,
                    SIGNAL_STORAGE: 8,
                    WORKER_STORAGE: 9,
                    SW: 10,
                    WAWC: 11,
                    WAWC_DB_ENC: 12,
                },
            ],
            webDbVersionNumber: [2, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'SyncdKeyCount',
        id: 3978,
        props: {
            keysUsedInSnapshotCount: [1, 'integer'],
            p80MuationsPerKey: [2, 'integer'],
            p95MuationsPerKey: [3, 'integer'],
            syncdSessionLengthDays: [4, 'integer'],
            totalKeyCount: [5, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MdAppStateSyncMutationStats',
        id: 3180,
        props: {
            applied: [
                1,
                {
                    ZERO: 1,
                    ONE: 2,
                    LT10: 3,
                    LT100: 4,
                    LT500: 5,
                    LT1K: 6,
                    LT5K: 7,
                    GTE5K: 8,
                },
            ],
            failed: [
                2,
                {
                    ZERO: 1,
                    ONE: 2,
                    LT10: 3,
                    LT100: 4,
                    LT500: 5,
                    LT1K: 6,
                    LT5K: 7,
                    GTE5K: 8,
                },
            ],
            invalid: [
                3,
                {
                    ZERO: 1,
                    ONE: 2,
                    LT10: 3,
                    LT100: 4,
                    LT500: 5,
                    LT1K: 6,
                    LT5K: 7,
                    GTE5K: 8,
                },
            ],
            orphan: [
                4,
                {
                    ZERO: 1,
                    ONE: 2,
                    LT10: 3,
                    LT100: 4,
                    LT500: 5,
                    LT1K: 6,
                    LT5K: 7,
                    GTE5K: 8,
                },
            ],
            syncdAction: [5, 'string'],
            unsupported: [
                6,
                {
                    ZERO: 1,
                    ONE: 2,
                    LT10: 3,
                    LT100: 4,
                    LT500: 5,
                    LT1K: 6,
                    LT5K: 7,
                    GTE5K: 8,
                },
            ],
        },
        weight: 20,
        wamChannel: 'private',
        privateStatsIdInt: 0,
    },
    {
        name: 'GroupJoinC',
        id: 158,
        props: {},
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MessageHighRetryCount',
        id: 3132,
        props: {
            deviceSizeBucket: [
                5,
                {
                    LT32: 1,
                    LT64: 2,
                    LT128: 3,
                    LT256: 4,
                    LT512: 5,
                    LT1000: 6,
                    LT1024: 16,
                    LT1500: 7,
                    LT2000: 8,
                    LT2500: 9,
                    LT3000: 10,
                    LT3500: 11,
                    LT4000: 12,
                    LT4500: 13,
                    LT5000: 14,
                    LARGEST_BUCKET: 15,
                },
            ],
            e2eSenderType: [
                3,
                {
                    MY_PRIMARY: 1,
                    OTHER_PRIMARY: 2,
                    MY_COMPANION: 3,
                    OTHER_COMPANION: 4,
                    MY_HOSTED_COMPANION: 5,
                    OTHER_HOSTED_COMPANION: 6,
                },
            ],
            mediaType: [
                1,
                {
                    NONE: 1,
                    PHOTO: 2,
                    VIDEO: 3,
                    AUDIO: 4,
                    PTT: 5,
                    LOCATION: 6,
                    CONTACT: 7,
                    DOCUMENT: 8,
                    URL: 9,
                    CALL: 10,
                    GIF: 11,
                    FUTURE: 12,
                    CONTACT_ARRAY: 13,
                    LIVE_LOCATION: 14,
                    PROFILE_PIC: 15,
                    STICKER: 16,
                    HSM: 17,
                    PRODUCT_IMAGE: 18,
                    TEMPLATE: 19,
                    MD_APP_STATE: 20,
                    MD_HISTORY_SYNC: 21,
                    CATALOG_LINK: 22,
                    PRODUCT_LINK: 23,
                    ORDER: 24,
                    LIST: 25,
                    LIST_REPLY: 26,
                    BUTTON_MESSAGE: 27,
                    BUTTON_RESPONSE_MESSAGE: 28,
                    PRODUCT_LIST: 29,
                    BUTTON_NFM: 30,
                    VIEW_ONCE_IMAGE: 31,
                    VIEW_ONCE_VIDEO: 32,
                    SHOP_STOREFRONT: 33,
                    REACTION: 34,
                    INTERACTIVE_NFM: 35,
                    INTERACTIVE_RESPONSE_NFM: 36,
                    POLL_CREATE: 37,
                    POLL_VOTE: 38,
                    TEXT: 39,
                    KEEP: 40,
                    UNKEEP: 41,
                    INVISIBLE_HELLO: 42,
                    MEDIA_EXPRESS_NOTIFY: 43,
                    UNDECRYPTED_MEDIA: 44,
                    UNDECRYPTED_TEXT: 45,
                    UNDECRYPTED_PAY: 46,
                    UNDECRYPTED_POLL: 47,
                    UNDECRYPTED_REACTION: 48,
                    UNDECRYPTED_MEDIA_NOTIFY: 49,
                    VIEW_ONCE_AUDIO: 50,
                    SCHEDULED_CALL_CREATE: 51,
                    SCHEDULED_CALL_CANCEL: 52,
                    PUSH_TO_VIDEO: 53,
                    PIN_IN_CHAT: 54,
                    INTERACTIVE_CAROUSEL: 55,
                    COMMENT: 56,
                    EPHEMERAL_SYNC_RESPONSE: 57,
                    EVENT_CREATE: 58,
                    EVENT_RESPOND: 59,
                    LOTTIE_STICKER: 60,
                    INTERACTIVE_PRODUCT_CAROUSEL: 61,
                    INTERACTIVE_PRODUCT: 62,
                },
            ],
            messageType: [
                4,
                {
                    INDIVIDUAL: 1,
                    GROUP: 2,
                    BROADCAST: 3,
                    STATUS: 4,
                    CHANNEL: 5,
                    INTEROP: 6,
                },
            ],
            retryCount: [2, 'integer'],
        },
        weight: 20,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'OfflineCountTooHigh',
        id: 2638,
        props: {
            callStanzaType: [
                7,
                {
                    OFFER: 0,
                    ACCEPT: 1,
                    REJECT: 2,
                    VIDEO: 3,
                    TERMINATE: 4,
                    ENC_REKEY: 5,
                    RELAYLATENCY: 6,
                    TRANSPORT: 7,
                    PREACCEPT: 8,
                    GROUP_UPDATE: 9,
                    MUTE_V2: 10,
                    INTERRUPTION: 11,
                    FLOWCONTROL: 12,
                    NOTIFY: 13,
                    OFFER_NOTICE: 14,
                    CALL_RELAY: 15,
                    MUTE: 16,
                    SCREEN_SHARE: 17,
                    UNKNOWN: 18,
                },
            ],
            mediaType: [
                3,
                {
                    NONE: 1,
                    PHOTO: 2,
                    VIDEO: 3,
                    AUDIO: 4,
                    PTT: 5,
                    LOCATION: 6,
                    CONTACT: 7,
                    DOCUMENT: 8,
                    URL: 9,
                    CALL: 10,
                    GIF: 11,
                    FUTURE: 12,
                    CONTACT_ARRAY: 13,
                    LIVE_LOCATION: 14,
                    PROFILE_PIC: 15,
                    STICKER: 16,
                    HSM: 17,
                    PRODUCT_IMAGE: 18,
                    TEMPLATE: 19,
                    MD_APP_STATE: 20,
                    MD_HISTORY_SYNC: 21,
                    CATALOG_LINK: 22,
                    PRODUCT_LINK: 23,
                    ORDER: 24,
                    LIST: 25,
                    LIST_REPLY: 26,
                    BUTTON_MESSAGE: 27,
                    BUTTON_RESPONSE_MESSAGE: 28,
                    PRODUCT_LIST: 29,
                    BUTTON_NFM: 30,
                    VIEW_ONCE_IMAGE: 31,
                    VIEW_ONCE_VIDEO: 32,
                    SHOP_STOREFRONT: 33,
                    REACTION: 34,
                    INTERACTIVE_NFM: 35,
                    INTERACTIVE_RESPONSE_NFM: 36,
                    POLL_CREATE: 37,
                    POLL_VOTE: 38,
                    TEXT: 39,
                    KEEP: 40,
                    UNKEEP: 41,
                    INVISIBLE_HELLO: 42,
                    MEDIA_EXPRESS_NOTIFY: 43,
                    UNDECRYPTED_MEDIA: 44,
                    UNDECRYPTED_TEXT: 45,
                    UNDECRYPTED_PAY: 46,
                    UNDECRYPTED_POLL: 47,
                    UNDECRYPTED_REACTION: 48,
                    UNDECRYPTED_MEDIA_NOTIFY: 49,
                    VIEW_ONCE_AUDIO: 50,
                    SCHEDULED_CALL_CREATE: 51,
                    SCHEDULED_CALL_CANCEL: 52,
                    PUSH_TO_VIDEO: 53,
                    PIN_IN_CHAT: 54,
                    INTERACTIVE_CAROUSEL: 55,
                    COMMENT: 56,
                    EPHEMERAL_SYNC_RESPONSE: 57,
                    EVENT_CREATE: 58,
                    EVENT_RESPOND: 59,
                    LOTTIE_STICKER: 60,
                    INTERACTIVE_PRODUCT_CAROUSEL: 61,
                    INTERACTIVE_PRODUCT: 62,
                },
            ],
            messageType: [
                4,
                {
                    INDIVIDUAL: 1,
                    GROUP: 2,
                    BROADCAST: 3,
                    STATUS: 4,
                    CHANNEL: 5,
                    INTEROP: 6,
                },
            ],
            notificationStanzaType: [6, 'string'],
            offlineCount: [2, 'integer'],
            receiptStanzaType: [5, 'string'],
            stanzaType: [
                1,
                {
                    MESSAGE: 1,
                    RECEIPT: 2,
                    CALL: 3,
                    NOTIFICATION: 4,
                    APPDATA: 5,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'E2eMessageRecv',
        id: 478,
        props: {
            agentEngagementType: [
                15,
                {
                    DIRECT_CHAT: 0,
                    INVOKED: 1,
                    MEMBER: 2,
                },
            ],
            botType: [
                19,
                {
                    UNKNOWN: 0,
                    METABOT: 1,
                    BOT_1P_BIZ: 2,
                    BOT_3P_BIZ: 3,
                },
            ],
            e2eCiphertextType: [
                5,
                {
                    MESSAGE: 0,
                    PREKEY_MESSAGE: 1,
                    SENDER_KEY_MESSAGE: 2,
                    MESSAGE_SECRET_MESSAGE: 3,
                },
            ],
            e2eCiphertextVersion: [6, 'integer'],
            e2eDestination: [
                4,
                {
                    INDIVIDUAL: 0,
                    GROUP: 1,
                    LIST: 2,
                    STATUS: 3,
                    CHANNEL: 4,
                    INTEROP: 5,
                },
            ],
            e2eFailureReason: [
                2,
                {
                    INVALID_MESSAGE: 0,
                    LEGACY_MESSAGE: 1,
                    INVALID_VERSION: 2,
                    PRE_KEY_MESSAGE_MISSING_PRE_KEY: 3,
                    PRE_KEY_MESSAGE_INVALID_KEY: 4,
                    UNTRUSTED_IDENTITY: 5,
                    NO_SESSION_AVAILABLE: 6,
                    PLAINTEXT_BYTE_MISMATCH: 7,
                    UNKNOWN_CIPHERTEXT_TYPE: 8,
                    UNKNOWN_CIPHERTEXT_VERSION: 9,
                    INVALID_PADDING: 10,
                    INVALID_PROTOCOL_BUFFER: 11,
                    INVALID_NUMBER_OF_MESSAGE_TYPES: 12,
                    INVALID_IMAGE_FILE_LENGTH: 13,
                    INVALID_IMAGE_FILE_SHA256: 14,
                    INVALID_IMAGE_MEDIA_URL: 15,
                    INVALID_IMAGE_MEDIA_KEY: 16,
                    INVALID_IMAGE_MIME_TYPE: 17,
                    UNKNOWN_V2_TAGS: 18,
                    INVALID_EDIT_VERSION: 19,
                    INVALID_HSM_NAMESPACE: 20,
                    INVALID_HSM_ELEMENT: 21,
                    UNKNOWN_PROTOCOL_MESSAGE_TYPE: 22,
                    DUPLICATE_MESSAGE: 23,
                    FINGERPRINT_IDENTITY_MISMATCH: 24,
                    FINGERPRINT_VERSION_MISMATCH: 25,
                    INVALID_ARGUMENT: 26,
                    INVALID_KEY: 27,
                    INVALID_KEY_ID: 28,
                    INVALID_MAC: 29,
                    NO_MEMORY: 30,
                    STALE_KEY_EXCHANGE: 31,
                    VRF_SIGNATURE_VERIFICATION_FAILED: 32,
                    ERROR_MINIMUM_VALUE: 33,
                    ERROR_UNKNOWN: 34,
                    INVALID_BROADCAST_STANZA_ATTRIBUTE: 35,
                    INVALID_LIST_REPLY_MESSAGE: 36,
                    INVALID_LIST_REPLY_ROW_IDENTIFIER: 37,
                    INVALID_BUTTONS_RESPONSE_MESSAGE: 38,
                    INVALID_BUTTONS_RESPONSE_BUTTON_IDENTIFIER: 39,
                    INVALID_ORDER_MESSAGE: 40,
                    ERROR_CANCELLED: 41,
                    FUTURE_MESSAGE: 42,
                    INVALID_EMPTY_PROTOBUF_NOT_ALLOWED: 43,
                    SELF_SEND: 44,
                    INVALID_MEDIA_MIME_TYPE_TEMPLATE_MESSAGE: 45,
                    INVALID_MEDIA_ENC_HASH: 46,
                    INVALID_IDENTITY: 47,
                    PASSTHROUGH: 48,
                    NOT_IN_PENDING_DEVICES: 49,
                    EDIT_TYPE_AND_PROTOCOL_MISMATCH: 50,
                    DECRYPTION_FAILED: 51,
                    DESERIALIZE_INVALID_PROTO_FORMAT: 52,
                    DESERIALIZE_RATCHET_KEY_BAD_FORMAT: 53,
                    DESERIALIZE_PK_INVALID_PROTO_FORMAT: 54,
                    DESERIALZE_PK_KEY_BAD_FORMAT: 55,
                    TOO_MANY_MESSAGES_IN_FUTURE: 56,
                    EMPTY_VERSION_CONTENT_SUFFIX: 57,
                    CONTENT_EXCEEDED_EXPECTED_LENGTH: 58,
                    GRP_INVALID_PROTO_FORMAT: 59,
                    GRP_INVALID_KEY_FORMAT: 60,
                    GRP_VERSION_CONTENT_EMPTY: 61,
                    GRP_INVALID_VERSION_CONTENT_LENGTH: 62,
                    GRP_SENDER_KEY_INVALID_PROTO_FORMAT: 63,
                    GRP_SENDER_KEY_PROTO_ERROR: 64,
                    GRP_TOO_MANY_MESSAGES_IN_FUTURE: 65,
                    MISSING_MESSAGE_SECRET: 66,
                    MESSAGE_SECRET_INVALID: 67,
                    INVALID_REACTION_TEXT: 68,
                    UNSUPPORTED_MESSAGE_TYPE: 69,
                    UNSUPPORTED_IN_INTEROP: 70,
                    TARGET_MESSAGE_NOT_FOUND: 71,
                    STATUS_FROM_HOSTED_DEVICE_UNSUPPORTED: 72,
                    GROUP_MESSAGE_FROM_HOSTED_DEVICE_UNSUPPORTED: 73,
                    BROADCAST_FROM_HOSTED_DEVICE_UNSUPPORTED: 74,
                    LIVE_LOCATION_FROM_HOSTED_DEVICE_UNSUPPORTED: 75,
                    MESSAGE_TYPE_MISMATCH: 76,
                    ERROR_SESSION_STATE_UNACKED_PKMSG_GET_BASE_KEY: 77,
                    ERROR_SIGNAL_MESSAGE_GET_SENDER_RATCHET_KEY: 78,
                    ERROR_SESSION_STATE_GET_ROOT_KEY: 79,
                    ERROR_SESSION_STATE_GET_SENDER_RATCHET_KEY_PAIR: 80,
                    ERROR_SESSION_STATE_SET_SENDER_CHAIN_KEY: 81,
                    ERROR_SESSION_STATE_FIND_RECEIVER_CHAIN: 82,
                    ERROR_SESSION_RECORD_DESERIALIZE: 83,
                    ERROR_SENDER_KEY_STATE_REMOVE_SENDER_MESSAGE_KEY: 84,
                    ERROR_EMPTY_PLAINTEXT_DATA: 85,
                    ERROR_MAC_LEN_MISMATCH: 86,
                    ERROR_INVALID_MAC_LEN: 87,
                    ERROR_INVALID_SIGNATURE_LEN: 88,
                    ERROR_SENDER_KEY_LOAD_KEY: 89,
                    ERROR_FAST_RATCHET_SENDER_KEY_LOAD_KEY: 90,
                    ERROR_UNKNOWN_LOAD_KEY: 91,
                    ERROR_UNKNOWN_LOAD_SESSION: 92,
                    ERROR_SESSION_RECORD_GET_STATE: 93,
                    ERROR_SESSION_STATE_GET_SENDER_CHAIN_KEY: 94,
                    ERROR_INVALID_KEY_MATEIRAL_DATA_LEN: 95,
                    ERROR_SESSION_STATE_GET_SENDER_RATCHET_KEY: 96,
                    ERROR_SESSION_STATE_GET_LOCAL_IDENTITY_KEY: 97,
                    ERROR_SESSION_STATE_GET_REMOTE_IDENTITY_KEY: 98,
                },
            ],
            e2eSenderType: [
                8,
                {
                    MY_PRIMARY: 1,
                    OTHER_PRIMARY: 2,
                    MY_COMPANION: 3,
                    OTHER_COMPANION: 4,
                    MY_HOSTED_COMPANION: 5,
                    OTHER_HOSTED_COMPANION: 6,
                },
            ],
            e2eSuccessful: [1, 'boolean'],
            editType: [
                13,
                {
                    NOT_EDITED: 0,
                    EDITED: 1,
                    SENDER_REVOKE: 2,
                    ADMIN_REVOKE: 3,
                },
            ],
            isHostedChat: [20, 'boolean'],
            isLid: [11, 'boolean'],
            localAddressingMode: [
                16,
                {
                    PN: 1,
                    LID: 2,
                },
            ],
            messageAddressingMode: [
                17,
                {
                    PN: 1,
                    LID: 2,
                },
            ],
            messageMediaType: [
                7,
                {
                    NONE: 1,
                    PHOTO: 2,
                    VIDEO: 3,
                    AUDIO: 4,
                    PTT: 5,
                    LOCATION: 6,
                    CONTACT: 7,
                    DOCUMENT: 8,
                    URL: 9,
                    CALL: 10,
                    GIF: 11,
                    FUTURE: 12,
                    CONTACT_ARRAY: 13,
                    LIVE_LOCATION: 14,
                    PROFILE_PIC: 15,
                    STICKER: 16,
                    HSM: 17,
                    PRODUCT_IMAGE: 18,
                    TEMPLATE: 19,
                    MD_APP_STATE: 20,
                    MD_HISTORY_SYNC: 21,
                    CATALOG_LINK: 22,
                    PRODUCT_LINK: 23,
                    ORDER: 24,
                    LIST: 25,
                    LIST_REPLY: 26,
                    BUTTON_MESSAGE: 27,
                    BUTTON_RESPONSE_MESSAGE: 28,
                    PRODUCT_LIST: 29,
                    BUTTON_NFM: 30,
                    VIEW_ONCE_IMAGE: 31,
                    VIEW_ONCE_VIDEO: 32,
                    SHOP_STOREFRONT: 33,
                    REACTION: 34,
                    INTERACTIVE_NFM: 35,
                    INTERACTIVE_RESPONSE_NFM: 36,
                    POLL_CREATE: 37,
                    POLL_VOTE: 38,
                    TEXT: 39,
                    KEEP: 40,
                    UNKEEP: 41,
                    INVISIBLE_HELLO: 42,
                    MEDIA_EXPRESS_NOTIFY: 43,
                    UNDECRYPTED_MEDIA: 44,
                    UNDECRYPTED_TEXT: 45,
                    UNDECRYPTED_PAY: 46,
                    UNDECRYPTED_POLL: 47,
                    UNDECRYPTED_REACTION: 48,
                    UNDECRYPTED_MEDIA_NOTIFY: 49,
                    VIEW_ONCE_AUDIO: 50,
                    SCHEDULED_CALL_CREATE: 51,
                    SCHEDULED_CALL_CANCEL: 52,
                    PUSH_TO_VIDEO: 53,
                    PIN_IN_CHAT: 54,
                    INTERACTIVE_CAROUSEL: 55,
                    COMMENT: 56,
                    EPHEMERAL_SYNC_RESPONSE: 57,
                    EVENT_CREATE: 58,
                    EVENT_RESPOND: 59,
                    LOTTIE_STICKER: 60,
                    INTERACTIVE_PRODUCT_CAROUSEL: 61,
                    INTERACTIVE_PRODUCT: 62,
                },
            ],
            offline: [9, 'boolean'],
            retryCount: [3, 'integer'],
            revokeType: [
                10,
                {
                    SENDER: 0,
                    ADMIN: 1,
                },
            ],
            serverAddressingMode: [
                18,
                {
                    PN: 1,
                    LID: 2,
                },
            ],
            stanzaType: [
                14,
                {
                    MESSAGE: 1,
                    RECEIPT: 2,
                    CALL: 3,
                    NOTIFICATION: 4,
                    APPDATA: 5,
                },
            ],
            typeOfGroup: [
                12,
                {
                    GROUP: 1,
                    SUBGROUP: 2,
                    DEFAULT_SUBGROUP: 3,
                },
            ],
        },
        weight: 20,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'ReceiptStanzaReceive',
        id: 2496,
        props: {
            dbReadsCount: [12, 'integer'],
            dbWritesCount: [13, 'integer'],
            mediaType: [
                2,
                {
                    NONE: 1,
                    PHOTO: 2,
                    VIDEO: 3,
                    AUDIO: 4,
                    PTT: 5,
                    LOCATION: 6,
                    CONTACT: 7,
                    DOCUMENT: 8,
                    URL: 9,
                    CALL: 10,
                    GIF: 11,
                    FUTURE: 12,
                    CONTACT_ARRAY: 13,
                    LIVE_LOCATION: 14,
                    PROFILE_PIC: 15,
                    STICKER: 16,
                    HSM: 17,
                    PRODUCT_IMAGE: 18,
                    TEMPLATE: 19,
                    MD_APP_STATE: 20,
                    MD_HISTORY_SYNC: 21,
                    CATALOG_LINK: 22,
                    PRODUCT_LINK: 23,
                    ORDER: 24,
                    LIST: 25,
                    LIST_REPLY: 26,
                    BUTTON_MESSAGE: 27,
                    BUTTON_RESPONSE_MESSAGE: 28,
                    PRODUCT_LIST: 29,
                    BUTTON_NFM: 30,
                    VIEW_ONCE_IMAGE: 31,
                    VIEW_ONCE_VIDEO: 32,
                    SHOP_STOREFRONT: 33,
                    REACTION: 34,
                    INTERACTIVE_NFM: 35,
                    INTERACTIVE_RESPONSE_NFM: 36,
                    POLL_CREATE: 37,
                    POLL_VOTE: 38,
                    TEXT: 39,
                    KEEP: 40,
                    UNKEEP: 41,
                    INVISIBLE_HELLO: 42,
                    MEDIA_EXPRESS_NOTIFY: 43,
                    UNDECRYPTED_MEDIA: 44,
                    UNDECRYPTED_TEXT: 45,
                    UNDECRYPTED_PAY: 46,
                    UNDECRYPTED_POLL: 47,
                    UNDECRYPTED_REACTION: 48,
                    UNDECRYPTED_MEDIA_NOTIFY: 49,
                    VIEW_ONCE_AUDIO: 50,
                    SCHEDULED_CALL_CREATE: 51,
                    SCHEDULED_CALL_CANCEL: 52,
                    PUSH_TO_VIDEO: 53,
                    PIN_IN_CHAT: 54,
                    INTERACTIVE_CAROUSEL: 55,
                    COMMENT: 56,
                    EPHEMERAL_SYNC_RESPONSE: 57,
                    EVENT_CREATE: 58,
                    EVENT_RESPOND: 59,
                    LOTTIE_STICKER: 60,
                    INTERACTIVE_PRODUCT_CAROUSEL: 61,
                    INTERACTIVE_PRODUCT: 62,
                },
            ],
            messageType: [
                10,
                {
                    INDIVIDUAL: 1,
                    GROUP: 2,
                    BROADCAST: 3,
                    STATUS: 4,
                    CHANNEL: 5,
                    INTEROP: 6,
                },
            ],
            receiptAggregation: [
                11,
                {
                    NONE: 0,
                    MULTI_MESSAGES: 1,
                    MULTI_PARTICIPANTS: 2,
                },
            ],
            receiptStanzaDuration: [1, 'timer'],
            receiptStanzaHasOrphaned: [6, 'boolean'],
            receiptStanzaOfflineCount: [3, 'integer'],
            receiptStanzaProcessedCount: [8, 'integer'],
            receiptStanzaRetryVer: [5, 'integer'],
            receiptStanzaStage: [
                9,
                {
                    OVERALL: 0,
                    PARSE: 1,
                    WAITING_TO_PROCESS: 2,
                    PROCESS: 3,
                    WAITING_TO_ACK: 5,
                    ACK: 4,
                },
            ],
            receiptStanzaTotalCount: [7, 'integer'],
            receiptStanzaType: [4, 'string'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'CtwaActionBannerUnderstand',
        id: 3586,
        props: {
            bannerIdentifier: [1, 'string'],
            bannerLocale: [2, 'string'],
            clientLocale: [3, 'string'],
            hasLocalLink: [4, 'boolean'],
            hasUniversalLink: [5, 'boolean'],
            invalidLink: [6, 'string'],
            lwiFlowIdentifier: [7, 'string'],
            preferredLink: [
                8,
                {
                    LOCAL: 0,
                    UNIVERSAL: 1,
                },
            ],
            validLocale: [9, 'boolean'],
            validNotification: [10, 'boolean'],
        },
        weight: 1,
        wamChannel: 'private',
        privateStatsIdInt: 0,
    },
    {
        name: 'WaOldCode',
        id: 3940,
        props: {
            deviceId: [1, 'string'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'NotificationDelivery',
        id: 3748,
        props: {
            clientMessageId: [1, 'string'],
            notificationAction: [
                2,
                {
                    SHOW: 1,
                    REMOVE: 2,
                },
            ],
            notificationDeliveryT: [3, 'integer'],
            notificationDestination: [
                4,
                {
                    INDIVIDUAL: 1,
                    GROUP: 2,
                    OTHER: 3,
                    CHANNEL: 4,
                    INTEROP: 5,
                },
            ],
            notificationId: [5, 'string'],
            notificationSource: [
                6,
                {
                    PUSH_TRIGGERED: 1,
                    MAIN_APP: 2,
                    IN_APP: 3,
                },
            ],
            threadId: [7, 'string'],
            uiNotificationType: [
                8,
                {
                    TEXT_MESSAGE: 1,
                    VOICE_MESSAGE: 2,
                    FILE_MESSAGE: 3,
                    PHOTO_MESSAGE: 4,
                    VIDEO_MESSAGE: 5,
                    MISSED_VOICE_CALL: 6,
                    MISSED_VIDEO_CALL: 7,
                    STICKER_MESSAGE: 8,
                    REACTION_MESSAGE: 9,
                    GROUP_INVITE_MESSAGE: 10,
                    REQUEST_MEDIA_REUPLOAD: 11,
                    WEB_CLIENT_LOGIN: 12,
                    BACKUP: 13,
                    KEEP_IN_CHAT: 14,
                    PAYMENT: 15,
                    POLL_MESSAGE: 16,
                    LOCATION_MESSAGE: 17,
                    WEB_PAGE_MESSAGE: 18,
                    FAILED_UPLOAD: 19,
                    GDPR_REPORT: 20,
                    INCOMING_VOICE_CALL: 21,
                    INCOMING_VIDEO_CALL: 22,
                    VIDEO_CALL_UPGRADE: 23,
                    WAFFLE: 24,
                    XPLAT_MIGRATION: 25,
                    OTHER: 26,
                    INVITE_JOINED: 27,
                    SCHEDULED_CALL_LOCAL_REMINDER: 28,
                    PTV_MESSAGE: 29,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MdAppStateDirtyBits',
        id: 2520,
        props: {
            dirtyBitsFalsePositive: [2, 'boolean'],
        },
        weight: 1000,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MdAppStateOfflineNotifications',
        id: 2602,
        props: {
            redundantCount: [1, 'integer'],
        },
        weight: 1000,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MdRetryFromUnknownDevice',
        id: 2178,
        props: {
            offline: [2, 'boolean'],
            senderType: [
                1,
                {
                    PRIMARY: 1,
                    COMPANION: 2,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'E2eRetryReject',
        id: 3578,
        props: {
            messageType: [
                1,
                {
                    INDIVIDUAL: 1,
                    GROUP: 2,
                    BROADCAST: 3,
                    STATUS: 4,
                    CHANNEL: 5,
                    INTEROP: 6,
                },
            ],
            msgRetryCount: [2, 'integer'],
            retryRejectReason: [
                3,
                {
                    OTHER: 0,
                    DOUBLE_CHECKMARK: 1,
                    IDENTITY_CHANGE: 2,
                    MESSAGE_NOT_EXIST: 3,
                },
            ],
            retryRevoke: [4, 'boolean'],
            senderDeviceType: [
                5,
                {
                    PRIMARY: 1,
                    COMPANION: 2,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'ClockSkewDifferenceT',
        id: 3178,
        props: {
            clockSkewHourly: [1, 'integer'],
        },
        weight: 10000,
        wamChannel: 'private',
        privateStatsIdInt: 37887164,
    },
    {
        name: 'MdBootstrapAppStateCriticalDataProcessing',
        id: 3164,
        props: {
            bootstrapAppStateDataStage: [
                1,
                {
                    REQUEST_BUILT: 1,
                    RESPONSE_RECEIVED: 2,
                    RESPONSE_PARSED_VALID: 3,
                    MISSING_KEYS_REQUESTED: 4,
                    MISSING_KEYS_RECEIVED: 5,
                    MUTATIONS_DECRYPTED: 6,
                    ABOUT_TO_APPLY_MUTATIONS: 7,
                    APPLIED_MUTATIONS: 8,
                    PUSHNAME_APPLIED: 9,
                    PUSHNAME_INVALID: 10,
                    ENTERED_RETRY_MODE: 11,
                },
            ],
            mdBootstrapPayloadType: [
                2,
                {
                    CRITICAL: 1,
                    NON_CRITICAL: 2,
                },
            ],
            mdRegAttemptId: [3, 'string'],
            mdSessionId: [4, 'string'],
            mdTimestamp: [5, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebcNativeUpsellCta',
        id: 3934,
        props: {
            webcNativeUpsellCtaEventType: [
                2,
                {
                    IMPRESSION: 1,
                    CTA_BTN_CLICK: 2,
                    CTA_DISMISS: 3,
                },
            ],
            webcNativeUpsellCtaSource: [
                1,
                {
                    INTRO_PANEL: 1,
                    CHATLIST_DROPDOWN: 2,
                    BUTTERBAR: 3,
                    QR_BANNER: 4,
                    SEARCH_RESULTS: 5,
                    CALL_BTN_MODAL: 6,
                    CALL_BTN_MODAL_2: 7,
                    MISSED_CALL_MODAL: 8,
                    MISSED_CALL_MODAL_2: 9,
                    QR_BANNER_2: 10,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebDbTableUsage',
        id: 5074,
        props: {
            offlineSessionId: [1, 'string'],
            webScenario: [
                2,
                {
                    OFFLINE_RESUME: 0,
                },
            ],
            webTable: [3, 'string'],
            webTableLogReason: [
                4,
                {
                    BASE: 0,
                    EXCEEDED_THRESHOLD: 1,
                },
            ],
            webTableReadCount: [5, 'integer'],
            webTableWriteCount: [6, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebDbLoadFromVersionFailureNonAnonymous',
        id: 4814,
        props: {
            webDbLoader: [
                1,
                {
                    MAIN: 1,
                    WEB_WORKER: 2,
                    SERVICE_WORKER: 3,
                },
            ],
            webDbName: [
                2,
                {
                    MODEL_STORAGE: 1,
                    FTS_STORAGE: 2,
                    JOBS_STORAGE: 3,
                    LOADGEN_STORAGE: 4,
                    LRU_MEDIA_STORAGE_IDB: 5,
                    OFFD_STORAGE: 6,
                    QPL_STORAGE: 7,
                    SIGNAL_STORAGE: 8,
                    WORKER_STORAGE: 9,
                    SW: 10,
                    WAWC: 11,
                    WAWC_DB_ENC: 12,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebcJobInfo',
        id: 3054,
        props: {
            jobName: [1, 'string'],
            jobPriority: [2, 'string'],
            jobResultType: [
                5,
                {
                    COMPLETED: 0,
                    ERROR: 1,
                    TIMEOUT: 2,
                    ABORTED: 3,
                },
            ],
            pendingJobsCount: [4, 'integer'],
            scenario: [
                3,
                {
                    INITIAL_PAIRING: 0,
                    OFFLINE_RESUME: 1,
                    IDLE: 2,
                },
            ],
            webcJobAddedT: [6, 'integer'],
            webcJobCompletedT: [8, 'integer'],
            webcJobStartedT: [7, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebcRawPlatforms',
        id: 2416,
        props: {
            webcRawPlatform: [1, 'string'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebWamForceFlush',
        id: 3264,
        props: {},
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'ChatMute',
        id: 2280,
        props: {
            actionConducted: [
                3,
                {
                    MUTE: 1,
                    UNMUTE: 2,
                    EXPIRE: 3,
                },
            ],
            chatMuteNotificationChoice: [
                5,
                {
                    NO_NOTIFICATIONS_WHEN_MUTED: 1,
                    YES_NOTIFICATIONS_WHEN_MUTED: 2,
                },
            ],
            muteChatType: [
                4,
                {
                    ONE_ON_ONE: 1,
                    GROUP: 2,
                    CHANNEL: 3,
                    INTEROP: 4,
                },
            ],
            muteDuration: [1, 'timer'],
            muteEntryPoint: [
                6,
                {
                    CHAT_LIST_SCREEN: 1,
                    CONTACT_INFO: 2,
                    CONVERSATION_SCREEN: 3,
                },
            ],
            muteGroupSize: [2, 'integer'],
            waOfficialAccountName: [
                7,
                {
                    WHATSAPP_CHATPSA: 1,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'ChatPsaAction',
        id: 3572,
        props: {
            messageMediaType: [
                1,
                {
                    NONE: 1,
                    PHOTO: 2,
                    VIDEO: 3,
                    AUDIO: 4,
                    PTT: 5,
                    LOCATION: 6,
                    CONTACT: 7,
                    DOCUMENT: 8,
                    URL: 9,
                    CALL: 10,
                    GIF: 11,
                    FUTURE: 12,
                    CONTACT_ARRAY: 13,
                    LIVE_LOCATION: 14,
                    PROFILE_PIC: 15,
                    STICKER: 16,
                    HSM: 17,
                    PRODUCT_IMAGE: 18,
                    TEMPLATE: 19,
                    MD_APP_STATE: 20,
                    MD_HISTORY_SYNC: 21,
                    CATALOG_LINK: 22,
                    PRODUCT_LINK: 23,
                    ORDER: 24,
                    LIST: 25,
                    LIST_REPLY: 26,
                    BUTTON_MESSAGE: 27,
                    BUTTON_RESPONSE_MESSAGE: 28,
                    PRODUCT_LIST: 29,
                    BUTTON_NFM: 30,
                    VIEW_ONCE_IMAGE: 31,
                    VIEW_ONCE_VIDEO: 32,
                    SHOP_STOREFRONT: 33,
                    REACTION: 34,
                    INTERACTIVE_NFM: 35,
                    INTERACTIVE_RESPONSE_NFM: 36,
                    POLL_CREATE: 37,
                    POLL_VOTE: 38,
                    TEXT: 39,
                    KEEP: 40,
                    UNKEEP: 41,
                    INVISIBLE_HELLO: 42,
                    MEDIA_EXPRESS_NOTIFY: 43,
                    UNDECRYPTED_MEDIA: 44,
                    UNDECRYPTED_TEXT: 45,
                    UNDECRYPTED_PAY: 46,
                    UNDECRYPTED_POLL: 47,
                    UNDECRYPTED_REACTION: 48,
                    UNDECRYPTED_MEDIA_NOTIFY: 49,
                    VIEW_ONCE_AUDIO: 50,
                    SCHEDULED_CALL_CREATE: 51,
                    SCHEDULED_CALL_CANCEL: 52,
                    PUSH_TO_VIDEO: 53,
                    PIN_IN_CHAT: 54,
                    INTERACTIVE_CAROUSEL: 55,
                    COMMENT: 56,
                    EPHEMERAL_SYNC_RESPONSE: 57,
                    EVENT_CREATE: 58,
                    EVENT_RESPOND: 59,
                    LOTTIE_STICKER: 60,
                    INTERACTIVE_PRODUCT_CAROUSEL: 61,
                    INTERACTIVE_PRODUCT: 62,
                },
            ],
            psaCampaignId: [4, 'string'],
            psaMessageActionType: [
                2,
                {
                    SAVE: 1,
                    FORWARD: 2,
                    REACT: 3,
                    LINK_CLICK: 4,
                    MEDIA_PLAY: 5,
                    DELETE: 6,
                },
            ],
            psaMsgId: [5, 'string'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'ChatPsaRead',
        id: 3574,
        props: {
            messageMediaType: [
                1,
                {
                    NONE: 1,
                    PHOTO: 2,
                    VIDEO: 3,
                    AUDIO: 4,
                    PTT: 5,
                    LOCATION: 6,
                    CONTACT: 7,
                    DOCUMENT: 8,
                    URL: 9,
                    CALL: 10,
                    GIF: 11,
                    FUTURE: 12,
                    CONTACT_ARRAY: 13,
                    LIVE_LOCATION: 14,
                    PROFILE_PIC: 15,
                    STICKER: 16,
                    HSM: 17,
                    PRODUCT_IMAGE: 18,
                    TEMPLATE: 19,
                    MD_APP_STATE: 20,
                    MD_HISTORY_SYNC: 21,
                    CATALOG_LINK: 22,
                    PRODUCT_LINK: 23,
                    ORDER: 24,
                    LIST: 25,
                    LIST_REPLY: 26,
                    BUTTON_MESSAGE: 27,
                    BUTTON_RESPONSE_MESSAGE: 28,
                    PRODUCT_LIST: 29,
                    BUTTON_NFM: 30,
                    VIEW_ONCE_IMAGE: 31,
                    VIEW_ONCE_VIDEO: 32,
                    SHOP_STOREFRONT: 33,
                    REACTION: 34,
                    INTERACTIVE_NFM: 35,
                    INTERACTIVE_RESPONSE_NFM: 36,
                    POLL_CREATE: 37,
                    POLL_VOTE: 38,
                    TEXT: 39,
                    KEEP: 40,
                    UNKEEP: 41,
                    INVISIBLE_HELLO: 42,
                    MEDIA_EXPRESS_NOTIFY: 43,
                    UNDECRYPTED_MEDIA: 44,
                    UNDECRYPTED_TEXT: 45,
                    UNDECRYPTED_PAY: 46,
                    UNDECRYPTED_POLL: 47,
                    UNDECRYPTED_REACTION: 48,
                    UNDECRYPTED_MEDIA_NOTIFY: 49,
                    VIEW_ONCE_AUDIO: 50,
                    SCHEDULED_CALL_CREATE: 51,
                    SCHEDULED_CALL_CANCEL: 52,
                    PUSH_TO_VIDEO: 53,
                    PIN_IN_CHAT: 54,
                    INTERACTIVE_CAROUSEL: 55,
                    COMMENT: 56,
                    EPHEMERAL_SYNC_RESPONSE: 57,
                    EVENT_CREATE: 58,
                    EVENT_RESPOND: 59,
                    LOTTIE_STICKER: 60,
                    INTERACTIVE_PRODUCT_CAROUSEL: 61,
                    INTERACTIVE_PRODUCT: 62,
                },
            ],
            psaCampaignId: [4, 'string'],
            psaMsgId: [5, 'string'],
            readEntryPoint: [
                3,
                {
                    CHAT_LIST: 1,
                    CHAT: 2,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'ChatPsaRemove',
        id: 3582,
        props: {
            lastReceivedMediaType: [
                2,
                {
                    NONE: 1,
                    PHOTO: 2,
                    VIDEO: 3,
                    AUDIO: 4,
                    PTT: 5,
                    LOCATION: 6,
                    CONTACT: 7,
                    DOCUMENT: 8,
                    URL: 9,
                    CALL: 10,
                    GIF: 11,
                    FUTURE: 12,
                    CONTACT_ARRAY: 13,
                    LIVE_LOCATION: 14,
                    PROFILE_PIC: 15,
                    STICKER: 16,
                    HSM: 17,
                    PRODUCT_IMAGE: 18,
                    TEMPLATE: 19,
                    MD_APP_STATE: 20,
                    MD_HISTORY_SYNC: 21,
                    CATALOG_LINK: 22,
                    PRODUCT_LINK: 23,
                    ORDER: 24,
                    LIST: 25,
                    LIST_REPLY: 26,
                    BUTTON_MESSAGE: 27,
                    BUTTON_RESPONSE_MESSAGE: 28,
                    PRODUCT_LIST: 29,
                    BUTTON_NFM: 30,
                    VIEW_ONCE_IMAGE: 31,
                    VIEW_ONCE_VIDEO: 32,
                    SHOP_STOREFRONT: 33,
                    REACTION: 34,
                    INTERACTIVE_NFM: 35,
                    INTERACTIVE_RESPONSE_NFM: 36,
                    POLL_CREATE: 37,
                    POLL_VOTE: 38,
                    TEXT: 39,
                    KEEP: 40,
                    UNKEEP: 41,
                    INVISIBLE_HELLO: 42,
                    MEDIA_EXPRESS_NOTIFY: 43,
                    UNDECRYPTED_MEDIA: 44,
                    UNDECRYPTED_TEXT: 45,
                    UNDECRYPTED_PAY: 46,
                    UNDECRYPTED_POLL: 47,
                    UNDECRYPTED_REACTION: 48,
                    UNDECRYPTED_MEDIA_NOTIFY: 49,
                    VIEW_ONCE_AUDIO: 50,
                    SCHEDULED_CALL_CREATE: 51,
                    SCHEDULED_CALL_CANCEL: 52,
                    PUSH_TO_VIDEO: 53,
                    PIN_IN_CHAT: 54,
                    INTERACTIVE_CAROUSEL: 55,
                    COMMENT: 56,
                    EPHEMERAL_SYNC_RESPONSE: 57,
                    EVENT_CREATE: 58,
                    EVENT_RESPOND: 59,
                    LOTTIE_STICKER: 60,
                    INTERACTIVE_PRODUCT_CAROUSEL: 61,
                    INTERACTIVE_PRODUCT: 62,
                },
            ],
            lastReceivedMessageTs: [4, 'timer'],
            lastReceivedMsgId: [8, 'string'],
            psaBlockReason: [
                10,
                {
                    OTHER: 0,
                    MESSAGES_ARENT_HELPFUL: 1,
                    TOO_MANY_MESSAGES: 2,
                    IT_LOOKS_SUSPICIOUS: 3,
                },
            ],
            psaCampaignId: [9, 'string'],
            psaMessageRemoveAction: [
                5,
                {
                    BLOCK: 1,
                    UNBLOCK: 2,
                    ARCHIVE: 3,
                    UNARCHIVE: 4,
                    CLEAR: 5,
                    DELETE_ALL: 6,
                },
            ],
            psaMessageRemoveEntryPoint: [
                6,
                {
                    BLOCK_FROM_CONTACT_INFO: 1,
                    BLOCK_FROM_CHAT: 2,
                    UNBLOCK_FROM_CONTACT_INFO: 3,
                    UNBLOCK_FROM_CHAT: 4,
                    UNBLOCK_FROM_PRIVACY_SETTINGS: 5,
                    ARCHIVE_FROM_CHAT_LIST: 6,
                    ARCHIVE_FROM_DELETE_OPTION: 7,
                    UNARCHIVE_FROM_ARCHIVED_CHAT_LIST: 8,
                    CLEAR_FROM_CONTACT_INFO: 9,
                    CLEAR_FROM_CHAT_LIST: 10,
                    DELETE_ALL_FROM_CHAT_LIST: 11,
                    DELETE_ALL_FROM_CONTACT_INFO: 12,
                    DELETE_ALL_FROM_CONVERSATION: 13,
                    CLEAR_FROM_CONVERSATION: 14,
                    BLOCK_FROM_CONSENT_MODAL: 15,
                },
            ],
            waOfficialAccountName: [
                7,
                {
                    WHATSAPP_CHATPSA: 1,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'ForwardSend',
        id: 1728,
        props: {
            disappearingChatInitiator: [
                21,
                {
                    CHAT: 1,
                    INITIATED_BY_ME: 2,
                    INITIATED_BY_OTHER: 3,
                    CHAT_PICKER: 4,
                    BIZ_UPGRADE_FB_HOSTING: 5,
                },
            ],
            e2eCiphertextType: [
                12,
                {
                    MESSAGE: 0,
                    PREKEY_MESSAGE: 1,
                    SENDER_KEY_MESSAGE: 2,
                    MESSAGE_SECRET_MESSAGE: 3,
                },
            ],
            e2eCiphertextVersion: [11, 'integer'],
            ephemeralityDuration: [18, 'integer'],
            ephemeralityInitiator: [
                24,
                {
                    INITIATED_BY_ME: 1,
                    INITIATED_BY_OTHER: 2,
                    BIZ_UPGRADE_FB_HOSTING: 3,
                },
            ],
            ephemeralityTriggerAction: [
                25,
                {
                    UNKNOWN: 0,
                    CHAT_SETTINGS: 1,
                    ACCOUNT_SETTINGS: 2,
                    BULK_CHANGE: 3,
                    BIZ_SUPPORTS_FB_HOSTING: 4,
                },
            ],
            fastForwardEnabled: [5, 'boolean'],
            isForwardedForward: [22, 'boolean'],
            isFrequentlyForwarded: [14, 'boolean'],
            mediaCaptionPresent: [10, 'boolean'],
            messageBizType: [
                26,
                {
                    NOT_BIZ_MSG: 0,
                    API_MARKETING: 1,
                    API_UTILITY: 2,
                    OTHER_API_BIZ_MSG: 3,
                    SMB_BIZ_MSG: 4,
                },
            ],
            messageForwardAgeT: [4, 'timer'],
            messageIsFanout: [6, 'boolean'],
            messageIsFastForward: [3, 'boolean'],
            messageIsInternational: [9, 'boolean'],
            messageMediaType: [
                2,
                {
                    NONE: 1,
                    PHOTO: 2,
                    VIDEO: 3,
                    AUDIO: 4,
                    PTT: 5,
                    LOCATION: 6,
                    CONTACT: 7,
                    DOCUMENT: 8,
                    URL: 9,
                    CALL: 10,
                    GIF: 11,
                    FUTURE: 12,
                    CONTACT_ARRAY: 13,
                    LIVE_LOCATION: 14,
                    PROFILE_PIC: 15,
                    STICKER: 16,
                    HSM: 17,
                    PRODUCT_IMAGE: 18,
                    TEMPLATE: 19,
                    MD_APP_STATE: 20,
                    MD_HISTORY_SYNC: 21,
                    CATALOG_LINK: 22,
                    PRODUCT_LINK: 23,
                    ORDER: 24,
                    LIST: 25,
                    LIST_REPLY: 26,
                    BUTTON_MESSAGE: 27,
                    BUTTON_RESPONSE_MESSAGE: 28,
                    PRODUCT_LIST: 29,
                    BUTTON_NFM: 30,
                    VIEW_ONCE_IMAGE: 31,
                    VIEW_ONCE_VIDEO: 32,
                    SHOP_STOREFRONT: 33,
                    REACTION: 34,
                    INTERACTIVE_NFM: 35,
                    INTERACTIVE_RESPONSE_NFM: 36,
                    POLL_CREATE: 37,
                    POLL_VOTE: 38,
                    TEXT: 39,
                    KEEP: 40,
                    UNKEEP: 41,
                    INVISIBLE_HELLO: 42,
                    MEDIA_EXPRESS_NOTIFY: 43,
                    UNDECRYPTED_MEDIA: 44,
                    UNDECRYPTED_TEXT: 45,
                    UNDECRYPTED_PAY: 46,
                    UNDECRYPTED_POLL: 47,
                    UNDECRYPTED_REACTION: 48,
                    UNDECRYPTED_MEDIA_NOTIFY: 49,
                    VIEW_ONCE_AUDIO: 50,
                    SCHEDULED_CALL_CREATE: 51,
                    SCHEDULED_CALL_CANCEL: 52,
                    PUSH_TO_VIDEO: 53,
                    PIN_IN_CHAT: 54,
                    INTERACTIVE_CAROUSEL: 55,
                    COMMENT: 56,
                    EPHEMERAL_SYNC_RESPONSE: 57,
                    EVENT_CREATE: 58,
                    EVENT_RESPOND: 59,
                    LOTTIE_STICKER: 60,
                    INTERACTIVE_PRODUCT_CAROUSEL: 61,
                    INTERACTIVE_PRODUCT: 62,
                },
            ],
            messageSendT: [13, 'timer'],
            messageType: [
                1,
                {
                    INDIVIDUAL: 1,
                    GROUP: 2,
                    BROADCAST: 3,
                    STATUS: 4,
                    CHANNEL: 5,
                    INTEROP: 6,
                },
            ],
            receiverDefaultDisappearingDuration: [20, 'integer'],
            resendCount: [8, 'integer'],
            retryCount: [7, 'integer'],
            senderDefaultDisappearingDuration: [19, 'integer'],
            typeOfGroup: [
                23,
                {
                    GROUP: 1,
                    SUBGROUP: 2,
                    DEFAULT_SUBGROUP: 3,
                },
            ],
            wouldBeFrequentlyForwardedAt3: [16, 'boolean'],
            wouldBeFrequentlyForwardedAt4: [17, 'boolean'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'NonMessagePeerDataMediaUpload',
        id: 3902,
        props: {
            peerDataErrorCount: [1, 'integer'],
            peerDataExistingDataNoUploadCount: [2, 'integer'],
            peerDataNotFoundCount: [3, 'integer'],
            peerDataRequestCount: [4, 'integer'],
            peerDataRequestSessionId: [5, 'string'],
            peerDataRequestType: [
                6,
                {
                    UPLOAD_STICKER: 0,
                    SEND_RECENT_STICKER_BOOTSTRAP: 1,
                    GENERAL_LINK_PREVIEW: 2,
                    HISTORY_SYNC_ON_DEMAND: 3,
                    PLACEHOLDER_MESSAGE_RESEND: 4,
                },
            ],
            peerDataResponseResult: [
                8,
                {
                    SUCCESS: 1,
                    OTHER_ERROR: 2,
                    REQUEST_TARGET_NOT_FOUND: 3,
                    REQUEST_INVALID: 4,
                    FAIL_TO_UPLOAD: 5,
                    FAIL_TO_SEND_RESPONSE: 6,
                    REQUEST_TOO_OLD: 7,
                },
            ],
            peerDataSuccessInlineNoUploadCount: [9, 'integer'],
            peerDataSuccessUploadCount: [7, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'NonMessagePeerDataOperationResponse',
        id: 3904,
        props: {
            peerDataErrorCount: [1, 'integer'],
            peerDataNotFoundCount: [2, 'integer'],
            peerDataRequestSessionId: [3, 'string'],
            peerDataRequestType: [
                4,
                {
                    UPLOAD_STICKER: 0,
                    SEND_RECENT_STICKER_BOOTSTRAP: 1,
                    GENERAL_LINK_PREVIEW: 2,
                    HISTORY_SYNC_ON_DEMAND: 3,
                    PLACEHOLDER_MESSAGE_RESEND: 4,
                },
            ],
            peerDataResponseApplyResult: [
                8,
                {
                    SUCCESS: 1,
                    OTHER_ERROR: 2,
                    INVALID_RESPONSE: 3,
                    FAIL_TO_DOWNLOAD: 4,
                    REQUEST_TIMEOUT: 5,
                },
            ],
            peerDataResponseCount: [5, 'integer'],
            peerDataSuccessProcessCount: [6, 'integer'],
            peerDataSuccessResponseCount: [7, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'NonMessagePeerDataRequest',
        id: 3906,
        props: {
            peerDataRequestCount: [1, 'integer'],
            peerDataRequestSessionId: [2, 'string'],
            peerDataRequestType: [
                3,
                {
                    UPLOAD_STICKER: 0,
                    SEND_RECENT_STICKER_BOOTSTRAP: 1,
                    GENERAL_LINK_PREVIEW: 2,
                    HISTORY_SYNC_ON_DEMAND: 3,
                    PLACEHOLDER_MESSAGE_RESEND: 4,
                },
            ],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'WebcLinkPreviewResponseHandle',
        id: 3860,
        props: {
            didRespondHqPreview: [5, 'boolean'],
            isPreviewSuccess: [2, 'boolean'],
            previewDurationMs: [4, 'integer'],
            previewSessionId: [3, 'string'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MdBootstrapHistoryDataDownloaded',
        id: 2296,
        props: {
            historySyncChunkOrder: [15, 'integer'],
            historySyncStageProgress: [14, 'integer'],
            mdBootstrapChatsCount: [6, 'integer'],
            mdBootstrapHistoryPayloadType: [
                13,
                {
                    INITIAL: 1,
                    RECENT_HISTORY: 2,
                    FULL_HISTORY: 3,
                    PUSHNAME: 4,
                    STATUS_V3: 5,
                    NON_BLOCKING_DATA: 6,
                    ON_DEMAND: 7,
                },
            ],
            mdBootstrapMessagesCount: [5, 'integer'],
            mdBootstrapPayloadSize: [4, 'integer'],
            mdBootstrapPayloadType: [
                2,
                {
                    CRITICAL: 1,
                    NON_CRITICAL: 2,
                },
            ],
            mdBootstrapStepDuration: [7, 'integer'],
            mdBootstrapStepResult: [
                8,
                {
                    SUCCESS: 1,
                    FAILURE: 2,
                },
            ],
            mdHsOldestMessageTimestamp: [11, 'integer'],
            mdRegAttemptId: [12, 'string'],
            mdSessionId: [1, 'string'],
            mdStorageQuotaBytes: [9, 'integer'],
            mdStorageQuotaUsedBytes: [10, 'integer'],
            mdTimestamp: [3, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'MdBootstrapHistoryDataStartDownloading',
        id: 4650,
        props: {
            historySyncChunkOrder: [1, 'integer'],
            historySyncStageProgress: [2, 'integer'],
            mdBootstrapHistoryPayloadType: [
                3,
                {
                    INITIAL: 1,
                    RECENT_HISTORY: 2,
                    FULL_HISTORY: 3,
                    PUSHNAME: 4,
                    STATUS_V3: 5,
                    NON_BLOCKING_DATA: 6,
                    ON_DEMAND: 7,
                },
            ],
            mdBootstrapPayloadSize: [4, 'integer'],
            mdBootstrapPayloadType: [
                5,
                {
                    CRITICAL: 1,
                    NON_CRITICAL: 2,
                },
            ],
            mdBootstrapStepDuration: [6, 'integer'],
            mdSessionId: [7, 'string'],
            mdTimestamp: [8, 'integer'],
        },
        weight: 1,
        wamChannel: 'regular',
        privateStatsIdInt: -1,
    },
    {
        name: 'E2eMessageSend',
        id: 476,
        props: {
            agentEngagementType: [
                15,
                {
                    DIRECT_CHAT: 0,
                    INVOKED: 1,
                    MEMBER: 2,
                },
            ],
            botType: [
                17,
                {
                    UNKNOWN: 0,
                    METABOT: 1,
                    BOT_1P_BIZ: 2,
                    BOT_3P_BIZ: 3,
                },
            ],
            e2eCiphertextType: [
                5,
                {
                    MESSAGE: 0,
                    PREKEY_MESSAGE: 1,
                    SENDER_KEY_MESSAGE: 2,
                    MESSAGE_SECRET_MESSAGE: 3,
                },
            ],
            e2eCiphertextVersion: [6, 'integer'],
            e2eDestination: [
                4,
                {
                    INDIVIDUAL: 0,
                    GROUP: 1,
                    LIST: 2,
                    STATUS: 3,
                    CHANNEL: 4,
                    INTEROP: 5,
                },
            ],
            e2eFailureReason: [
                2,
                {
                    INVALID_MESSAGE: 0,
                    LEGACY_MESSAGE: 1,
                    INVALID_VERSION: 2,
                    PRE_KEY_MESSAGE_MISSING_PRE_KEY: 3,
                    PRE_KEY_MESSAGE_INVALID_KEY: 4,
                    UNTRUSTED_IDENTITY: 5,
                    NO_SESSION_AVAILABLE: 6,
                    PLAINTEXT_BYTE_MISMATCH: 7,
                    UNKNOWN_CIPHERTEXT_TYPE: 8,
                    UNKNOWN_CIPHERTEXT_VERSION: 9,
                    INVALID_PADDING: 10,
                    INVALID_PROTOCOL_BUFFER: 11,
                    INVALID_NUMBER_OF_MESSAGE_TYPES: 12,
                    INVALID_IMAGE_FILE_LENGTH: 13,
                    INVALID_IMAGE_FILE_SHA256: 14,
                    INVALID_IMAGE_MEDIA_URL: 15,
                    INVALID_IMAGE_MEDIA_KEY: 16,
                    INVALID_IMAGE_MIME_TYPE: 17,
                    UNKNOWN_V2_TAGS: 18,
                    INVALID_EDIT_VERSION: 19,
                    INVALID_HSM_NAMESPACE: 20,
                    INVALID_HSM_ELEMENT: 21,
                    UNKNOWN_PROTOCOL_MESSAGE_TYPE: 22,
                    DUPLICATE_MESSAGE: 23,
                    FINGERPRINT_IDENTITY_MISMATCH: 24,
                    FINGERPRINT_VERSION_MISMATCH: 25,
                    INVALID_ARGUMENT: 26,
                    INVALID_KEY: 27,
                    INVALID_KEY_ID: 28,
                    INVALID_MAC: 29,
                    NO_MEMORY: 30,
                    STALE_KEY_EXCHANGE: 31,
                    VRF_SIGNATURE_VERIFICATION_FAILED: 32,
                    ERROR_MINIMUM_VALUE: 33,
                    ERROR_UNKNOWN: 34,
                    INVALID_BROADCAST_STANZA_ATTRIBUTE: 35,
                    INVALID_LIST_REPLY_MESSAGE: 36,
                    INVALID_LIST_REPLY_ROW_IDENTIFIER: 37,
                    INVALID_BUTTONS_RESPONSE_MESSAGE: 38,
                    INVALID_BUTTONS_RESPONSE_BUTTON_IDENTIFIER: 39,
                    INVALID_ORDER_MESSAGE: 40,
                    ERROR_CANCELLED: 41,
                    FUTURE_MESSAGE: 42,
                    INVALID_EMPTY_PROTOBUF_NOT_ALLOWED: 43,
                    SELF_SEND: 44,
                    INVALID_MEDIA_MIME_TYPE_TEMPLATE_MESSAGE: 45,
                    INVALID_MEDIA_ENC_HASH: 46,
                    INVALID_IDENTITY: 47,
                    PASSTHROUGH: 48,
                    NOT_IN_PENDING_DEVICES: 49,
                    EDIT_TYPE_AND_PROTOCOL_MISMATCH: 50,
                    DECRYPTION_FAILED: 51,
                    DESERIALIZE_INVALID_PROTO_FORMAT: 52,
                    DESERIALIZE_RATCHET_KEY_BAD_FORMAT: 53,
                    DESERIALIZE_PK_INVALID_PROTO_FORMAT: 54,
                    DESERIALZE_PK_KEY_BAD_FORMAT: 55,
                    TOO_MANY_MESSAGES_IN_FUTURE: 56,
                    EMPTY_VERSION_CONTENT_SUFFIX: 57,
                    CONTENT_EXCEEDED_EXPECTED_LENGTH: 58,
                    GRP_INVALID_PROTO_FORMAT: 59,
                    GRP_INVALID_KEY_FORMAT: 60,
                    GRP_VERSION_CONTENT_EMPTY: 61,
                    GRP_INVALID_VERSION_CONTENT_LENGTH: 62,
                    GRP_SENDER_KEY_INVALID_PROTO_FORMAT: 63,
                    GRP_SENDER_KEY_PROTO_ERROR: 64,
                    GRP_TOO_MANY_MESSAGES_IN_FUTURE: 65,
                    MISSING_MESSAGE_SECRET: 66,
                    MESSAGE_SECRET_INVALID: 67,
                    INVALID_REACTION_TEXT: 68,
                    UNSUPPORTED_MESSAGE_TYPE: 69,
                    UNSUPPORTED_IN_INTEROP: 70,
                    TARGET_MESSAGE_NOT_FOUND: 71,
                    STATUS_FROM_HOSTED_DEVICE_UNSUPPORTED: 72,
                    GROUP_MESSAGE_FROM_HOSTED_DEVICE_UNSUPPORTED: 73,
                    BROADCAST_FROM_HOSTED_DEVICE_UNSUPPORTED: 74,
                    LIVE_LOCATION_FROM_HOSTED_DEVICE_UNSUPPORTED: 75,
                    MESSAGE_TYPE_MISMATCH: 76,
                    ERROR_SESSION_STATE_UNACKED_PKMSG_GET_BASE_KEY: 77,
                    ERROR_SIGNAL_MESSAGE_GET_SENDER_RATCHET_KEY: 78,
                    ERROR_SESSION_STATE_GET_ROOT_KEY: 79,
                    ERROR_SESSION_STATE_GET_SENDER_RATCHET_KEY_PAIR: 80,
                    ERROR_SESSION_STATE_SET_SENDER_CHAIN_KEY: 81,
                    ERROR_SESSION_STATE_FIND_RECEIVER_CHAIN: 82,
                    ERROR_SESSION_RECORD_DESERIALIZE: 83,
                    ERROR_SENDER_KEY_STATE_REMOVE_SENDER_MESSAGE_KEY: 84,
                    ERROR_EMPTY_PLAINTEXT_DATA: 85,
                    ERROR_MAC_LEN_MISMATCH: 86,
                    ERROR_INVALID_MAC_LEN: 87,
                    ERROR_INVALID_SIGNATURE_LEN: 88,
                    ERROR_SENDER_KEY_LOAD_KEY: 89,
                    ERROR_FAST_RATCHET_SENDER_KEY_LOAD_KEY: 90,
                    ERROR_UNKNOWN_LOAD_KEY: 91,
                    ERROR_UNKNOWN_LOAD_SESSION: 92,
                    ERROR_SESSION_RECORD_GET_STATE: 93,
                    ERROR_SESSION_STATE_GET_SENDER_CHAIN_KEY: 94,
                    ERROR_INVALID_KEY_MATEIRAL_DATA_LEN: 95,
                    ERROR_SESSION_STATE_GET_SENDER_RATCHET_KEY: 96,
                    ERROR_SESSION_STATE_GET_LOCAL_IDENTITY_KEY: 97,
                    ERROR_SESSION_STATE_GET_REMOTE_IDENTITY_KEY: 98,
                },
            ],
            e2eReceiverType: [
                8,
                {
                    PRIMARY: 1,
                    COMPANION: 2,
                },
            ],
            e2eSuccessful: [1, 'boolean'],
            editType: [
                14,
                {
                    NOT_EDITED: 0,
                    EDITED: 1,
                    SENDER_REVOKE: 2,
                    ADMIN_REVOKE: 3,
                },
            ],
            encRetryCount: [9, 'integer'],
            isLid: [12, 'boolean'],
            localAddressingMode: [
                16,
                {
                    PN: 1,
                    LID: 2,
                },
            ],
            messageIsInvisible: [10, 'boolean'],
            messageMediaType: [
                7,
                {
                    NONE: 1,
                    PHOTO: 2,
                    VIDEO: 3,
                    AUDIO: 4,
                    PTT: 5,
                    LOCATION: 6,
                    CONTACT: 7,
                    DOCUMENT: 8,
                    URL: 9,
                    CALL: 10,
                    GIF: 11,
                    FUTURE: 12,
                    CONTACT_ARRAY: 13,
                    LIVE_LOCATION: 14,
                    PROFILE_PIC: 15,
                    STICKER: 16,
                    HSM: 17,
